<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 20 Adding calibration points to your phylogenies | Molecular Phylogenetic Techniques</title>
  <meta name="description" content="Description about Course/Book." />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 20 Adding calibration points to your phylogenies | Molecular Phylogenetic Techniques" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Description about Course/Book." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 20 Adding calibration points to your phylogenies | Molecular Phylogenetic Techniques" />
  
  <meta name="twitter:description" content="Description about Course/Book." />
  



<meta name="date" content="2022-07-25" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="assets/anvil_favicon.ico" type="image/x-icon" />
<link rel="prev" href="what-is-a-time-tree.html"/>
<link rel="next" href="about-the-authors.html"/>
<script src="libs/header-attrs-2.10/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="assets/AnVIL_style/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<a href="https://anvilproject.org/" target="_blank"><img src="assets/AnVIL_style/logo-anvil.png" style="width: 80%; padding-left: 15px; padding-top: 8px;"</a>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>About this Book</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#anvil-collection"><i class="fa fa-check"></i>AnVIL Collection</a></li>
</ul></li>
<li class="part"><span><b>BOOK OVERVIEW</b></span></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#motivation"><i class="fa fa-check"></i><b>1.1</b> Motivation</a></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#target-audience"><i class="fa fa-check"></i><b>1.2</b> Target Audience</a></li>
<li class="chapter" data-level="1.3" data-path="introduction.html"><a href="introduction.html#curriculum"><i class="fa fa-check"></i><b>1.3</b> Curriculum</a></li>
</ul></li>
<li class="part"><span><b>BEFORE YOU START</b></span></li>
<li class="chapter" data-level="2" data-path="sign-in.html"><a href="sign-in.html"><i class="fa fa-check"></i><b>2</b> Sign in</a></li>
<li class="chapter" data-level="3" data-path="create-a-workspace.html"><a href="create-a-workspace.html"><i class="fa fa-check"></i><b>3</b> Create a workspace</a></li>
<li class="part"><span><b>RUNNING RSTUDIO</b></span></li>
<li class="chapter" data-level="4" data-path="rstudio.html"><a href="rstudio.html"><i class="fa fa-check"></i><b>4</b> Working in RStudio</a>
<ul>
<li class="chapter" data-level="4.1" data-path="rstudio.html"><a href="rstudio.html#launch-rstudio-cloud-environment"><i class="fa fa-check"></i><b>4.1</b> Launch RStudio Cloud Environment</a></li>
<li class="chapter" data-level="4.2" data-path="rstudio.html"><a href="rstudio.html#tour-rstudio"><i class="fa fa-check"></i><b>4.2</b> Tour RStudio</a></li>
<li class="chapter" data-level="4.3" data-path="rstudio.html"><a href="rstudio.html#stopping"><i class="fa fa-check"></i><b>4.3</b> Pause RStudio</a></li>
<li class="chapter" data-level="4.4" data-path="rstudio.html"><a href="rstudio.html#delete-rstudio-cloud-environment"><i class="fa fa-check"></i><b>4.4</b> Delete RStudio Cloud Environment</a></li>
<li class="chapter" data-level="4.5" data-path="rstudio.html"><a href="rstudio.html#video-guide"><i class="fa fa-check"></i><b>4.5</b> Video Guide</a></li>
</ul></li>
<li class="part"><span><b>TREE THINKING</b></span></li>
<li class="chapter" data-level="5" data-path="phylogenetics-basics.html"><a href="phylogenetics-basics.html"><i class="fa fa-check"></i><b>5</b> Phylogenetics basics</a>
<ul>
<li class="chapter" data-level="5.1" data-path="phylogenetics-basics.html"><a href="phylogenetics-basics.html#how-to-read-a-phylogenetic-tree"><i class="fa fa-check"></i><b>5.1</b> How to read a phylogenetic tree</a></li>
<li class="chapter" data-level="5.2" data-path="phylogenetics-basics.html"><a href="phylogenetics-basics.html#outgroups"><i class="fa fa-check"></i><b>5.2</b> Outgroups</a></li>
<li class="chapter" data-level="5.3" data-path="phylogenetics-basics.html"><a href="phylogenetics-basics.html#branch-lengths"><i class="fa fa-check"></i><b>5.3</b> Branch lengths</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="visualizing-trees-in-r.html"><a href="visualizing-trees-in-r.html"><i class="fa fa-check"></i><b>6</b> Visualizing trees in R</a>
<ul>
<li class="chapter" data-level="6.1" data-path="visualizing-trees-in-r.html"><a href="visualizing-trees-in-r.html#creating-a-newick-object"><i class="fa fa-check"></i><b>6.1</b> Creating a Newick object</a></li>
<li class="chapter" data-level="6.2" data-path="visualizing-trees-in-r.html"><a href="visualizing-trees-in-r.html#drawing-trees"><i class="fa fa-check"></i><b>6.2</b> Drawing trees</a></li>
<li class="chapter" data-level="6.3" data-path="visualizing-trees-in-r.html"><a href="visualizing-trees-in-r.html#adding-outgroups"><i class="fa fa-check"></i><b>6.3</b> Adding outgroups</a></li>
<li class="chapter" data-level="6.4" data-path="visualizing-trees-in-r.html"><a href="visualizing-trees-in-r.html#drawing-trees-multiple-ways"><i class="fa fa-check"></i><b>6.4</b> Drawing trees multiple ways</a></li>
<li class="chapter" data-level="6.5" data-path="visualizing-trees-in-r.html"><a href="visualizing-trees-in-r.html#the-phylo-class"><i class="fa fa-check"></i><b>6.5</b> The <code>phylo</code> class</a></li>
</ul></li>
<li class="part"><span><b>GETTING SEQUENCES</b></span></li>
<li class="chapter" data-level="7" data-path="what-sequences-should-i-choose.html"><a href="what-sequences-should-i-choose.html"><i class="fa fa-check"></i><b>7</b> What sequences should I choose?</a></li>
<li class="chapter" data-level="8" data-path="finding-sequences-in-genbank.html"><a href="finding-sequences-in-genbank.html"><i class="fa fa-check"></i><b>8</b> Finding sequences in GenBank</a>
<ul>
<li class="chapter" data-level="8.1" data-path="finding-sequences-in-genbank.html"><a href="finding-sequences-in-genbank.html#identifying-a-query-sequence-on-genbank"><i class="fa fa-check"></i><b>8.1</b> Identifying a query sequence on GenBank</a></li>
<li class="chapter" data-level="8.2" data-path="finding-sequences-in-genbank.html"><a href="finding-sequences-in-genbank.html#blastn"><i class="fa fa-check"></i><b>8.2</b> blastn</a></li>
<li class="chapter" data-level="8.3" data-path="finding-sequences-in-genbank.html"><a href="finding-sequences-in-genbank.html#identifying-an-outgroup"><i class="fa fa-check"></i><b>8.3</b> Identifying an outgroup</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="downloading-the-sequences-from-genbank.html"><a href="downloading-the-sequences-from-genbank.html"><i class="fa fa-check"></i><b>9</b> Downloading the sequences from GenBank</a>
<ul>
<li class="chapter" data-level="9.1" data-path="downloading-the-sequences-from-genbank.html"><a href="downloading-the-sequences-from-genbank.html#the-dna.bin-object"><i class="fa fa-check"></i><b>9.1</b> The DNA.bin object</a></li>
<li class="chapter" data-level="9.2" data-path="downloading-the-sequences-from-genbank.html"><a href="downloading-the-sequences-from-genbank.html#the-fasta-format"><i class="fa fa-check"></i><b>9.2</b> The fasta format</a></li>
</ul></li>
<li class="part"><span><b>ALIGNMENT</b></span></li>
<li class="chapter" data-level="10" data-path="why-do-we-align-sequences.html"><a href="why-do-we-align-sequences.html"><i class="fa fa-check"></i><b>10</b> Why do we align sequences?</a></li>
<li class="chapter" data-level="11" data-path="creating-a-multiple-sequence-alignment.html"><a href="creating-a-multiple-sequence-alignment.html"><i class="fa fa-check"></i><b>11</b> Creating a multiple sequence alignment</a>
<ul>
<li class="chapter" data-level="11.1" data-path="creating-a-multiple-sequence-alignment.html"><a href="creating-a-multiple-sequence-alignment.html#installing-the-decipher-package"><i class="fa fa-check"></i><b>11.1</b> Installing the <code>DECIPHER</code> package</a></li>
<li class="chapter" data-level="11.2" data-path="creating-a-multiple-sequence-alignment.html"><a href="creating-a-multiple-sequence-alignment.html#loading-the-fasta-file"><i class="fa fa-check"></i><b>11.2</b> Loading the fasta file</a></li>
<li class="chapter" data-level="11.3" data-path="creating-a-multiple-sequence-alignment.html"><a href="creating-a-multiple-sequence-alignment.html#creating-an-alignment"><i class="fa fa-check"></i><b>11.3</b> Creating an alignment</a></li>
</ul></li>
<li class="part"><span><b>NEIGHBOR JOINING</b></span></li>
<li class="chapter" data-level="12" data-path="the-neighbor-joining-method-of-tree-estimation.html"><a href="the-neighbor-joining-method-of-tree-estimation.html"><i class="fa fa-check"></i><b>12</b> The neighbor joining method of tree estimation</a>
<ul>
<li class="chapter" data-level="12.1" data-path="the-neighbor-joining-method-of-tree-estimation.html"><a href="the-neighbor-joining-method-of-tree-estimation.html#models-of-molecular-evolution"><i class="fa fa-check"></i><b>12.1</b> Models of molecular evolution</a>
<ul>
<li class="chapter" data-level="12.1.1" data-path="the-neighbor-joining-method-of-tree-estimation.html"><a href="the-neighbor-joining-method-of-tree-estimation.html#jc69"><i class="fa fa-check"></i><b>12.1.1</b> JC69</a></li>
<li class="chapter" data-level="12.1.2" data-path="the-neighbor-joining-method-of-tree-estimation.html"><a href="the-neighbor-joining-method-of-tree-estimation.html#k2p"><i class="fa fa-check"></i><b>12.1.2</b> K2P</a></li>
<li class="chapter" data-level="12.1.3" data-path="the-neighbor-joining-method-of-tree-estimation.html"><a href="the-neighbor-joining-method-of-tree-estimation.html#gtr"><i class="fa fa-check"></i><b>12.1.3</b> GTR</a></li>
<li class="chapter" data-level="12.1.4" data-path="the-neighbor-joining-method-of-tree-estimation.html"><a href="the-neighbor-joining-method-of-tree-estimation.html#other-parameters"><i class="fa fa-check"></i><b>12.1.4</b> Other parameters</a></li>
<li class="chapter" data-level="12.1.5" data-path="the-neighbor-joining-method-of-tree-estimation.html"><a href="the-neighbor-joining-method-of-tree-estimation.html#how-do-i-choose-a-model"><i class="fa fa-check"></i><b>12.1.5</b> How do I choose a model?</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="building-a-neighbor-joining-tree.html"><a href="building-a-neighbor-joining-tree.html"><i class="fa fa-check"></i><b>13</b> Building a neighbor joining tree</a>
<ul>
<li class="chapter" data-level="13.1" data-path="building-a-neighbor-joining-tree.html"><a href="building-a-neighbor-joining-tree.html#the-phydat-object"><i class="fa fa-check"></i><b>13.1</b> The phyDat object</a></li>
<li class="chapter" data-level="13.2" data-path="building-a-neighbor-joining-tree.html"><a href="building-a-neighbor-joining-tree.html#choosing-a-substitution-model"><i class="fa fa-check"></i><b>13.2</b> Choosing a substitution model</a></li>
<li class="chapter" data-level="13.3" data-path="building-a-neighbor-joining-tree.html"><a href="building-a-neighbor-joining-tree.html#building-the-neighbor-joining-tree"><i class="fa fa-check"></i><b>13.3</b> Building the neighbor joining tree</a></li>
<li class="chapter" data-level="13.4" data-path="building-a-neighbor-joining-tree.html"><a href="building-a-neighbor-joining-tree.html#visualizing-the-neighbor-joining-tree"><i class="fa fa-check"></i><b>13.4</b> Visualizing the neighbor joining tree</a></li>
<li class="chapter" data-level="13.5" data-path="building-a-neighbor-joining-tree.html"><a href="building-a-neighbor-joining-tree.html#saving-your-trees"><i class="fa fa-check"></i><b>13.5</b> Saving your trees</a></li>
</ul></li>
<li class="part"><span><b>PARSIMONY</b></span></li>
<li class="chapter" data-level="14" data-path="the-parsimony-method-of-tree-estimation.html"><a href="the-parsimony-method-of-tree-estimation.html"><i class="fa fa-check"></i><b>14</b> The parsimony method of tree estimation</a>
<ul>
<li class="chapter" data-level="14.1" data-path="the-parsimony-method-of-tree-estimation.html"><a href="the-parsimony-method-of-tree-estimation.html#parsimony"><i class="fa fa-check"></i><b>14.1</b> Parsimony</a></li>
<li class="chapter" data-level="14.2" data-path="the-parsimony-method-of-tree-estimation.html"><a href="the-parsimony-method-of-tree-estimation.html#tree-searching-methods"><i class="fa fa-check"></i><b>14.2</b> Tree searching methods</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="the-parsimony-method-of-tree-estimation.html"><a href="the-parsimony-method-of-tree-estimation.html#branch-and-bound"><i class="fa fa-check"></i><b>14.2.1</b> Branch and bound</a></li>
<li class="chapter" data-level="14.2.2" data-path="the-parsimony-method-of-tree-estimation.html"><a href="the-parsimony-method-of-tree-estimation.html#nearest-neighbor-interchange-nni"><i class="fa fa-check"></i><b>14.2.2</b> Nearest Neighbor Interchange (NNI)</a></li>
<li class="chapter" data-level="14.2.3" data-path="the-parsimony-method-of-tree-estimation.html"><a href="the-parsimony-method-of-tree-estimation.html#subtree-pruning-and-regrafting-spr"><i class="fa fa-check"></i><b>14.2.3</b> Subtree Pruning and Regrafting (SPR)</a></li>
<li class="chapter" data-level="14.2.4" data-path="the-parsimony-method-of-tree-estimation.html"><a href="the-parsimony-method-of-tree-estimation.html#star-decomposition"><i class="fa fa-check"></i><b>14.2.4</b> Star decomposition</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="building-a-parsimony-tree.html"><a href="building-a-parsimony-tree.html"><i class="fa fa-check"></i><b>15</b> Building a parsimony tree</a>
<ul>
<li class="chapter" data-level="15.1" data-path="building-a-parsimony-tree.html"><a href="building-a-parsimony-tree.html#editing-the-fasta-file"><i class="fa fa-check"></i><b>15.1</b> Editing the fasta file</a></li>
<li class="chapter" data-level="15.2" data-path="building-a-parsimony-tree.html"><a href="building-a-parsimony-tree.html#inferring-a-parsimony-tree"><i class="fa fa-check"></i><b>15.2</b> Inferring a parsimony tree</a></li>
<li class="chapter" data-level="15.3" data-path="building-a-parsimony-tree.html"><a href="building-a-parsimony-tree.html#parsimony-scores"><i class="fa fa-check"></i><b>15.3</b> Parsimony scores</a></li>
<li class="chapter" data-level="15.4" data-path="building-a-parsimony-tree.html"><a href="building-a-parsimony-tree.html#bootstrapping"><i class="fa fa-check"></i><b>15.4</b> Bootstrapping</a></li>
<li class="chapter" data-level="15.5" data-path="building-a-parsimony-tree.html"><a href="building-a-parsimony-tree.html#saving-files"><i class="fa fa-check"></i><b>15.5</b> Saving files</a></li>
</ul></li>
<li class="part"><span><b>MORE TREE THINKING</b></span></li>
<li class="chapter" data-level="16" data-path="more-phylogenetics-basics.html"><a href="more-phylogenetics-basics.html"><i class="fa fa-check"></i><b>16</b> More Phylogenetics Basics</a>
<ul>
<li class="chapter" data-level="16.1" data-path="more-phylogenetics-basics.html"><a href="more-phylogenetics-basics.html#tree-topology"><i class="fa fa-check"></i><b>16.1</b> Tree topology</a></li>
<li class="chapter" data-level="16.2" data-path="more-phylogenetics-basics.html"><a href="more-phylogenetics-basics.html#outgroups-1"><i class="fa fa-check"></i><b>16.2</b> Outgroups</a></li>
<li class="chapter" data-level="16.3" data-path="more-phylogenetics-basics.html"><a href="more-phylogenetics-basics.html#comparing-trees"><i class="fa fa-check"></i><b>16.3</b> Comparing trees</a></li>
</ul></li>
<li class="part"><span><b>MAXIMUM LIKELIHOOD</b></span></li>
<li class="chapter" data-level="17" data-path="what-is-maximum-likelihood.html"><a href="what-is-maximum-likelihood.html"><i class="fa fa-check"></i><b>17</b> What is Maximum Likelihood?</a></li>
<li class="chapter" data-level="18" data-path="estimating-a-tree-using-maximum-likelihood.html"><a href="estimating-a-tree-using-maximum-likelihood.html"><i class="fa fa-check"></i><b>18</b> Estimating a tree using Maximum Likelihood</a>
<ul>
<li class="chapter" data-level="18.1" data-path="estimating-a-tree-using-maximum-likelihood.html"><a href="estimating-a-tree-using-maximum-likelihood.html#creating-a-starting-tree"><i class="fa fa-check"></i><b>18.1</b> Creating a starting tree</a></li>
<li class="chapter" data-level="18.2" data-path="estimating-a-tree-using-maximum-likelihood.html"><a href="estimating-a-tree-using-maximum-likelihood.html#calculating-likelihood-score"><i class="fa fa-check"></i><b>18.2</b> Calculating likelihood score</a></li>
<li class="chapter" data-level="18.3" data-path="estimating-a-tree-using-maximum-likelihood.html"><a href="estimating-a-tree-using-maximum-likelihood.html#optimizing-the-maximum-likelihood-parameters"><i class="fa fa-check"></i><b>18.3</b> Optimizing the maximum likelihood parameters</a></li>
<li class="chapter" data-level="18.4" data-path="estimating-a-tree-using-maximum-likelihood.html"><a href="estimating-a-tree-using-maximum-likelihood.html#changing-the-model-for-your-ml-search"><i class="fa fa-check"></i><b>18.4</b> Changing the model for your ML search</a></li>
<li class="chapter" data-level="18.5" data-path="estimating-a-tree-using-maximum-likelihood.html"><a href="estimating-a-tree-using-maximum-likelihood.html#bootstrapping-the-ml-tree"><i class="fa fa-check"></i><b>18.5</b> Bootstrapping the ML tree</a></li>
<li class="chapter" data-level="18.6" data-path="estimating-a-tree-using-maximum-likelihood.html"><a href="estimating-a-tree-using-maximum-likelihood.html#saving-your-bootstrap-tree"><i class="fa fa-check"></i><b>18.6</b> Saving your bootstrap tree</a></li>
</ul></li>
<li class="part"><span><b>TIME TREES</b></span></li>
<li class="chapter" data-level="19" data-path="what-is-a-time-tree.html"><a href="what-is-a-time-tree.html"><i class="fa fa-check"></i><b>19</b> What is a time tree?</a>
<ul>
<li class="chapter" data-level="19.1" data-path="what-is-a-time-tree.html"><a href="what-is-a-time-tree.html#molecular-clocks"><i class="fa fa-check"></i><b>19.1</b> Molecular clocks</a></li>
<li class="chapter" data-level="19.2" data-path="what-is-a-time-tree.html"><a href="what-is-a-time-tree.html#calibration-points"><i class="fa fa-check"></i><b>19.2</b> Calibration points</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="adding-calibration-points-to-your-phylogenies.html"><a href="adding-calibration-points-to-your-phylogenies.html"><i class="fa fa-check"></i><b>20</b> Adding calibration points to your phylogenies</a>
<ul>
<li class="chapter" data-level="20.1" data-path="adding-calibration-points-to-your-phylogenies.html"><a href="adding-calibration-points-to-your-phylogenies.html#choosing-calibration-points"><i class="fa fa-check"></i><b>20.1</b> Choosing calibration points</a></li>
<li class="chapter" data-level="20.2" data-path="adding-calibration-points-to-your-phylogenies.html"><a href="adding-calibration-points-to-your-phylogenies.html#adding-calibration-points-to-internal-nodes-for-ml-trees"><i class="fa fa-check"></i><b>20.2</b> Adding calibration points to internal nodes for ML trees</a></li>
<li class="chapter" data-level="20.3" data-path="adding-calibration-points-to-your-phylogenies.html"><a href="adding-calibration-points-to-your-phylogenies.html#adding-calibration-points-for-bayesian-trees"><i class="fa fa-check"></i><b>20.3</b> Adding calibration points for Bayesian trees</a></li>
<li class="chapter" data-level="20.4" data-path="adding-calibration-points-to-your-phylogenies.html"><a href="adding-calibration-points-to-your-phylogenies.html#visualizing-the-time-trees-in-figtree"><i class="fa fa-check"></i><b>20.4</b> Visualizing the time trees in FigTree</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="about-the-authors.html"><a href="about-the-authors.html"><i class="fa fa-check"></i>About the Authors</a></li>
<li class="chapter" data-level="21" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>21</b> References</a></li>
<li class="divider"></li>
<p style="text-align:center;"> <a href="https://github.com/jhudsl/DaSL_Course_Template_Bookdown" target="blank" > This content was published with</a> <a href="https://bookdown.org/" target="blank"> bookdown by: </a> </p>
<p style="text-align:center;"> <a href="http://jhudatascience.org/"> The Johns Hopkins Data Science Lab </a></p>
<a href="http://jhudatascience.org/"><img src="https://jhudatascience.org/images/dasl.png" style=" width: 80%; padding-left: 40px; padding-top: 8px; vertical-align: top "</a>
<p style="text-align:center; font-size: 12px;"> <a href="https://github.com/rstudio4edu/rstudio4edu-book/"> Style adapted from: rstudio4edu-book </a> <a href ="https://creativecommons.org/licenses/by/2.0/"> (CC-BY 2.0) </a></p>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Molecular Phylogenetic Techniques</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<head>
  <meta name="viewport" content="width=device-width,minimum-scale=1.0,maximum-scale=10.0,initial-scale=1.0">
  <!--script src="https://kit.fontawesome.com/6a26f47516.js"></script-->
  <!--<script src="assets/hideOutput.js"></script>-->
  <link href="assets/AnvIL_style/style.css" rel="stylesheet">
</head>
        


<div class="hero-image-container"> 
  <img class= "hero-image" src="assets/AnVIL_style/dasl_thin_main_image.png">
</div>
<div id="adding-calibration-points-to-your-phylogenies" class="section level1" number="20">
<h1><span class="header-section-number">Chapter 20</span> Adding calibration points to your phylogenies</h1>
<p>This week we will be working with both RStudio and BEAST. Within RStudio, we will be using the <code>ape</code> package again.</p>
<div id="choosing-calibration-points" class="section level2" number="20.1">
<h2><span class="header-section-number">20.1</span> Choosing calibration points</h2>
<p>This week, you should choose at least one calibration point to add to your tree. You can pick any two taxa from a tree and figure out the estimated age of the divergence between them (or the estimated age of the most recent common ancestor between them). The website <a href="http://timetree.org/">TimeTree</a> has divergence estimates for many different taxa pairs.</p>
<p>When two taxa (usually species, but it could be taxa from higher taxonomic levels) are entered into TimeTree, the website first checks whether these taxa are in the TimeTree of Life database. If they are, the divergence estimate between the two (the age of the MRCA between them) is retrieved. If one or both of the taxa are <em>not</em> in the database, the website scans NCBI taxonomy to find the closest relative of the species and uses that as a proxy to find the MRCA for two taxa.</p>
<p>If your taxa are not on TimeTree, you will have to be creative in finding a calibration point. You might be able to find information in the literature. If you are working with viral samples, you can use the date the sample was collected as a tip calibration point. If nothing works and you are stuck, email me and I will send you the grass dataset to work with.</p>
<p>For the grass dataset, let’s use the divergence between bread wheat (<em>Triticum aestivum</em>) and crested wheatgrass (<em>Agropyron cristatum</em>). When we search TimeTree, we see several estimated divergence times.</p>
<p><img src="resources/images/11-divergence.png" title="Major point!! example image" alt="Major point!! example image" style="display: block; margin: auto;" /></p>
<p>TimeTree gives us two divergence estimates: a median estimate (with 95% confidence interval), and an adjusted time, which is quite a bit older than the median estimate. This is what TimeTree says about their adjusted time estimate: “Answer: Due to conflicting time estimates between studies, ancestral nodes can be assigned younger ages than their descendants. A smoothing technique is used to adjust these times so the resulting tree is ultrametric.” (An ultrametric tree is one where all the a rooted tree with edge lengths where all leaves are equidistant from the root. When trees have been scaled using calibration points, they often become ultrametric.)</p>
<p>In this case, the very youngest divergence estimate from TimeTree is 2.6 million years ago (MYA), while the oldest is 16.5 MYA. We will use these estimates going forward. We could have easily chosen the other estimates as well. There isn’t necessarily a right or wrong choice here. You just need to be upfront about the estimates you chose.</p>
</div>
<div id="adding-calibration-points-to-internal-nodes-for-ml-trees" class="section level2" number="20.2">
<h2><span class="header-section-number">20.2</span> Adding calibration points to internal nodes for ML trees</h2>
<p>When we look at the grass tree, we see the node that indicates the MRCA between crested wheatgrass and bread wheat (labeled wheat on the tree) is pretty deep within the tree.</p>
<p><img src="resources/images/11-grass_tree_mapped.png" title="Major point!! example image" alt="Major point!! example image" style="display: block; margin: auto;" /></p>
<p>If we trace the branches until we find the node where they connect, we can see it is the basal node of the tree (the node which connects the ingroup to the outgroup). In order to add a calibration date to our ML tree in R, we need to know how R has identified (or labeled) that particular node. You can either load your saved ML tree or create it again.</p>
<div class="sourceCode" id="cb136"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb136-1"><a href="adding-calibration-points-to-your-phylogenies.html#cb136-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ape)</span>
<span id="cb136-2"><a href="adding-calibration-points-to-your-phylogenies.html#cb136-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phangorn)</span>
<span id="cb136-3"><a href="adding-calibration-points-to-your-phylogenies.html#cb136-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-4"><a href="adding-calibration-points-to-your-phylogenies.html#cb136-4" aria-hidden="true" tabindex="-1"></a>grass.align <span class="ot">&lt;-</span> <span class="fu">read.phyDat</span>(<span class="st">&quot;grass_aligned-renamed.fasta&quot;</span>, <span class="at">format =</span> <span class="st">&quot;fasta&quot;</span>)</span>
<span id="cb136-5"><a href="adding-calibration-points-to-your-phylogenies.html#cb136-5" aria-hidden="true" tabindex="-1"></a>dist <span class="ot">&lt;-</span> <span class="fu">dist.ml</span>(grass.align)</span>
<span id="cb136-6"><a href="adding-calibration-points-to-your-phylogenies.html#cb136-6" aria-hidden="true" tabindex="-1"></a>nj.tree <span class="ot">&lt;-</span> <span class="fu">nj</span>(dist)</span>
<span id="cb136-7"><a href="adding-calibration-points-to-your-phylogenies.html#cb136-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-8"><a href="adding-calibration-points-to-your-phylogenies.html#cb136-8" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">pml</span>(nj.tree, <span class="at">data =</span> grass.align)</span>
<span id="cb136-9"><a href="adding-calibration-points-to-your-phylogenies.html#cb136-9" aria-hidden="true" tabindex="-1"></a>fitGTR.G <span class="ot">&lt;-</span> <span class="fu">update</span>(fit, <span class="at">model =</span> <span class="st">&quot;GTR&quot;</span>, <span class="at">k =</span> <span class="dv">4</span>)</span>
<span id="cb136-10"><a href="adding-calibration-points-to-your-phylogenies.html#cb136-10" aria-hidden="true" tabindex="-1"></a>fitGTR.G <span class="ot">&lt;-</span> <span class="fu">optim.pml</span>(fitGTR.G, <span class="at">model =</span> <span class="st">&quot;GTR&quot;</span>, <span class="at">optGamma =</span> T, <span class="at">rearrangement =</span> <span class="st">&quot;stochastic&quot;</span>, <span class="at">control =</span> <span class="fu">pml.control</span>(<span class="at">trace =</span> <span class="dv">0</span>))</span>
<span id="cb136-11"><a href="adding-calibration-points-to-your-phylogenies.html#cb136-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-12"><a href="adding-calibration-points-to-your-phylogenies.html#cb136-12" aria-hidden="true" tabindex="-1"></a>tree.root <span class="ot">&lt;-</span> <span class="fu">root</span>(fitGTR.G<span class="sc">$</span>tree, <span class="at">outgroup =</span> <span class="fu">c</span>(<span class="st">&#39;barley_D-hordein&#39;</span>,<span class="st">&#39;Siberian wild rye_D-hordein&#39;</span>))</span></code></pre></div>
<div class="sourceCode" id="cb137"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb137-1"><a href="adding-calibration-points-to-your-phylogenies.html#cb137-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(tree.root)</span>
<span id="cb137-2"><a href="adding-calibration-points-to-your-phylogenies.html#cb137-2" aria-hidden="true" tabindex="-1"></a><span class="fu">nodelabels</span>()</span></code></pre></div>
<p><img src="resources/images/11-grass_tree_nodes.png" title="Major point!! example image" alt="Major point!! example image" style="display: block; margin: auto;" /></p>
<p>The node that connects crested wheatgrass and wheat (and indicates the MRCA between the two) is node 21. We will use that information, as well as the estimated divergence times from TimeTree, to apply a timescale to the ML tree using the <code>chronos</code> command from the <code>ape</code> package. If you’d like to read more about it, the <code>ape</code> manual is <a href="https://cran.r-project.org/web/packages/ape/ape.pdf">here</a>.</p>
<p>The <code>chronos</code> command takes several arguments. The first is a lambda starting estimate (the smoothing parameter). The next argument is the model of rate substitution variation (how much should the mutation rate vary among the branches of the tree?). <code>chronos</code> can implement a strict clock (no variation, called “clock”), a correlated clock (where the variation is correlated between branches, but does exist, called “correlated”), a discrete clock (where estimates are discrete for each branch and requires multiple estimated parameters, called “discrete”), or a relaxed clock. The relaxed clock (called “relaxed”) allows for discrete mutation rates on the branches, but doesn’t involve estimating a different parameter for each branch. We’ll implement the discrete clock in this example. (The estimation is fairly quick, so you can easily implement all four and see how the resulting trees differ.)</p>
<p>The remaining arguments is the calibration data frame (which includes the actual calibration points). We use the command <code>makeChronosCalib</code> command to create the calibration data frame. This command takes the ML tree as the first argument.</p>
<div class="sourceCode" id="cb138"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb138-1"><a href="adding-calibration-points-to-your-phylogenies.html#cb138-1" aria-hidden="true" tabindex="-1"></a>calib <span class="ot">&lt;-</span> <span class="fu">makeChronosCalib</span>(tree.root, <span class="at">node =</span> <span class="dv">21</span>, <span class="at">age.min =</span> <span class="fl">2.6</span>, <span class="at">age.max =</span> <span class="fl">16.5</span>)</span>
<span id="cb138-2"><a href="adding-calibration-points-to-your-phylogenies.html#cb138-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb138-3"><a href="adding-calibration-points-to-your-phylogenies.html#cb138-3" aria-hidden="true" tabindex="-1"></a>calib</span></code></pre></div>
<pre><code>##   node age.min age.max soft.bounds
## 1   21     2.6    16.5       FALSE</code></pre>
<p>If we want to add more calibration points, we can use the <code>c</code> (concatenate) command. This tells R that a string of multiple entries are being entered at once.</p>
<div class="sourceCode" id="cb140"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb140-1"><a href="adding-calibration-points-to-your-phylogenies.html#cb140-1" aria-hidden="true" tabindex="-1"></a>calib2 <span class="ot">&lt;-</span> <span class="fu">makeChronosCalib</span>(tree.root, <span class="at">node =</span> <span class="fu">c</span>(<span class="st">&quot;21&quot;</span>, <span class="st">&quot;19&quot;</span>), <span class="at">age.min =</span> <span class="fu">c</span>(<span class="fl">2.6</span>,<span class="fl">3.0</span>), <span class="at">age.max =</span> <span class="fu">c</span>(<span class="fl">16.5</span>, <span class="fl">16.5</span>))</span>
<span id="cb140-2"><a href="adding-calibration-points-to-your-phylogenies.html#cb140-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb140-3"><a href="adding-calibration-points-to-your-phylogenies.html#cb140-3" aria-hidden="true" tabindex="-1"></a>calib2</span></code></pre></div>
<pre><code>##   node age.min age.max soft.bounds
## 1   21     2.6    16.5       FALSE
## 2   19     3.0    16.5       FALSE</code></pre>
<p>Creating the time tree, we simply combine the <code>chronos</code> and <code>makeChronosCalib</code> commands.</p>
<div class="sourceCode" id="cb142"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb142-1"><a href="adding-calibration-points-to-your-phylogenies.html#cb142-1" aria-hidden="true" tabindex="-1"></a>time.tree <span class="ot">&lt;-</span> <span class="fu">as.phylo</span>(tree.root)</span>
<span id="cb142-2"><a href="adding-calibration-points-to-your-phylogenies.html#cb142-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb142-3"><a href="adding-calibration-points-to-your-phylogenies.html#cb142-3" aria-hidden="true" tabindex="-1"></a>time.tree_dated <span class="ot">&lt;-</span> <span class="fu">chronos</span>(time.tree, <span class="at">lambda =</span> <span class="dv">1</span>, <span class="at">model =</span> <span class="st">&quot;discrete&quot;</span>, <span class="at">quiet =</span> <span class="cn">FALSE</span>, <span class="at">calibration =</span> calib, <span class="at">control =</span> <span class="fu">chronos.control</span>())</span></code></pre></div>
<pre><code>## 
## Setting initial dates...
## Fitting in progress... get a first set of estimates
##          (Penalised) log-lik = -4.110631 
## Optimising rates... frequencies... dates... -4.110631 
## Optimising rates... frequencies... dates... -3.795125 
## 
## log-Lik = -3.795125 
## PHIIC = 67.59</code></pre>
<p>Before we plot the tree, we need to do a bit of formatting of the branch lengths (called edge.length). R will be default save these values to 10 decimal places, which can make visualization a little tricky. We’re going to format them so they only have two decimal places, then plot them on top of the time tree.</p>
<div class="sourceCode" id="cb144"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb144-1"><a href="adding-calibration-points-to-your-phylogenies.html#cb144-1" aria-hidden="true" tabindex="-1"></a>time.tree_dated<span class="sc">$</span>edge.length <span class="ot">&lt;-</span> <span class="fu">round</span>(time.tree_dated<span class="sc">$</span>edge.length, <span class="at">digits =</span> <span class="dv">2</span>)</span>
<span id="cb144-2"><a href="adding-calibration-points-to-your-phylogenies.html#cb144-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb144-3"><a href="adding-calibration-points-to-your-phylogenies.html#cb144-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(time.tree_dated)</span>
<span id="cb144-4"><a href="adding-calibration-points-to-your-phylogenies.html#cb144-4" aria-hidden="true" tabindex="-1"></a><span class="fu">edgelabels</span>(time.tree_dated<span class="sc">$</span>edge.length, <span class="at">bg=</span><span class="st">&quot;black&quot;</span>, <span class="at">col=</span><span class="st">&quot;white&quot;</span>, <span class="at">font=</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="resources/images/11-time-trees_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
<p>Behold, you have a time tree. The branch lengths are now scaled to million of years, so you can estimate the age of MRCA between any two taxa on the tree. For example, based on the discrete clock model, our tree suggests the MRCA between mosquito grass and medusahead rye lived ~11 MYA (the length of the branch between mosquito grass and the node that denotes the common ancestor with medusahead rye).</p>
<p>Before you end your session, make sure you save your tree (and then download the .tre file to your desktop).</p>
<div class="sourceCode" id="cb145"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb145-1"><a href="adding-calibration-points-to-your-phylogenies.html#cb145-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.tree</span>(time.tree_dated, <span class="st">&quot;ml_grass_timetree.tre&quot;</span>)</span></code></pre></div>
</div>
<div id="adding-calibration-points-for-bayesian-trees" class="section level2" number="20.3">
<h2><span class="header-section-number">20.3</span> Adding calibration points for Bayesian trees</h2>
<p>We can add calibration points using the BEAUTi gui you worked with last week. In this case, we open BEAUTi and load our dataset. We have two choices at this point: we can choose the “Tip Dates” tab and specify the date each sample was taken (useful for viral trees, especially during an outbreak), or we can choose the “Clock Model” tab (if you do this, use a strict clock model unless you are certain your data are not clock-like).</p>
<p>If you’ve chosen to not use Tip Dates, we will add additional calibration points in the “Priors” tab. These instructions are taken from the tutorial you worked with last week.</p>
<p><img src="resources/images/11-beast_1.png" title="Major point!! example image" alt="Major point!! example image" style="display: block; margin: auto;" /></p>
<p><img src="resources/images/11-beast_2.png" title="Major point!! example image" alt="Major point!! example image" style="display: block; margin: auto;" /></p>
<p>The rest of your parameters (including your substitution model and priors) can be set to the same values you used last week. Follow the instructions from last week to get your final Bayesian tree.</p>
</div>
<div id="visualizing-the-time-trees-in-figtree" class="section level2" number="20.4">
<h2><span class="header-section-number">20.4</span> Visualizing the time trees in FigTree</h2>
<p>After you finish creating your trees, you can visualize both the ML and Bayesian tree using FigTree. Play around with the options to make the tree look nice - I recommend making the tip labels larger. Be sure to include your branch lengths! Those are now scaled to be the divergence times for both the ML and Bayesian trees.</p>
<p><img src="resources/images/11-grass_ml_figtree.png" title="Major point!! example image" alt="Major point!! example image" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb146"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb146-1"><a href="adding-calibration-points-to-your-phylogenies.html#cb146-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.0.2 (2020-06-22)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.3 LTS
## 
## Matrix products: default
## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] phangorn_2.5.5 ape_5.4-1     
## 
## loaded via a namespace (and not attached):
##  [1] igraph_1.2.6    Rcpp_1.0.8      knitr_1.33      magrittr_2.0.2 
##  [5] hms_0.5.3       lattice_0.20-41 R6_2.4.1        quadprog_1.5-8 
##  [9] rlang_0.4.10    fastmatch_1.1-0 highr_0.8       stringr_1.4.0  
## [13] tools_4.0.2     parallel_4.0.2  grid_4.0.2      nlme_3.1-149   
## [17] xfun_0.26       jquerylib_0.1.4 htmltools_0.5.0 ellipsis_0.3.1 
## [21] ottrpal_0.1.2   yaml_2.2.1      digest_0.6.25   tibble_3.0.3   
## [25] lifecycle_1.0.0 crayon_1.3.4    bookdown_0.24   Matrix_1.2-18  
## [29] readr_1.4.0     vctrs_0.3.4     fs_1.5.0        evaluate_0.14  
## [33] rmarkdown_2.10  stringi_1.5.3   compiler_4.0.2  pillar_1.4.6   
## [37] pkgconfig_2.0.3</code></pre>

</div>
</div>
<hr>
<center> 
  <div class="footer">
      All illustrations <a href="https://creativecommons.org/licenses/by/4.0/">CC-BY. </a>
      <br>
      All other materials <a href= "https://creativecommons.org/licenses/by/4.0/"> CC-BY </a> unless noted otherwise.
  </div>
</center>
            </section>

          </div>
        </div>
      </div>
<a href="what-is-a-time-tree.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="about-the-authors.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
