<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 13 Building a neighbor joining tree | Molecular Phylogenetic Techniques</title>
  <meta name="description" content="Description about Course/Book." />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 13 Building a neighbor joining tree | Molecular Phylogenetic Techniques" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Description about Course/Book." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 13 Building a neighbor joining tree | Molecular Phylogenetic Techniques" />
  
  <meta name="twitter:description" content="Description about Course/Book." />
  



<meta name="date" content="2022-06-27" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="assets/anvil_favicon.ico" type="image/x-icon" />
<link rel="prev" href="the-neighbor-joining-method-of-tree-estimation.html"/>
<link rel="next" href="the-parsimony-method-of-tree-estimation.html"/>
<script src="libs/header-attrs-2.10/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="assets/AnVIL_style/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<a href="https://anvilproject.org/" target="_blank"><img src="assets/AnVIL_style/logo-anvil.png" style="width: 80%; padding-left: 15px; padding-top: 8px;"</a>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>About this Book</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#anvil-collection"><i class="fa fa-check"></i>AnVIL Collection</a></li>
</ul></li>
<li class="part"><span><b>BOOK OVERVIEW</b></span></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#motivation"><i class="fa fa-check"></i><b>1.1</b> Motivation</a></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#target-audience"><i class="fa fa-check"></i><b>1.2</b> Target Audience</a></li>
<li class="chapter" data-level="1.3" data-path="introduction.html"><a href="introduction.html#curriculum"><i class="fa fa-check"></i><b>1.3</b> Curriculum</a></li>
</ul></li>
<li class="part"><span><b>BEFORE YOU START</b></span></li>
<li class="chapter" data-level="2" data-path="sign-in.html"><a href="sign-in.html"><i class="fa fa-check"></i><b>2</b> Sign in</a></li>
<li class="chapter" data-level="3" data-path="create-a-workspace.html"><a href="create-a-workspace.html"><i class="fa fa-check"></i><b>3</b> Create a workspace</a></li>
<li class="part"><span><b>RUNNING RSTUDIO</b></span></li>
<li class="chapter" data-level="4" data-path="rstudio.html"><a href="rstudio.html"><i class="fa fa-check"></i><b>4</b> Working in RStudio</a>
<ul>
<li class="chapter" data-level="4.1" data-path="rstudio.html"><a href="rstudio.html#launch-rstudio-cloud-environment"><i class="fa fa-check"></i><b>4.1</b> Launch RStudio Cloud Environment</a></li>
<li class="chapter" data-level="4.2" data-path="rstudio.html"><a href="rstudio.html#tour-rstudio"><i class="fa fa-check"></i><b>4.2</b> Tour RStudio</a></li>
<li class="chapter" data-level="4.3" data-path="rstudio.html"><a href="rstudio.html#stopping"><i class="fa fa-check"></i><b>4.3</b> Pause RStudio</a></li>
<li class="chapter" data-level="4.4" data-path="rstudio.html"><a href="rstudio.html#delete-rstudio-cloud-environment"><i class="fa fa-check"></i><b>4.4</b> Delete RStudio Cloud Environment</a></li>
<li class="chapter" data-level="4.5" data-path="rstudio.html"><a href="rstudio.html#video-guide"><i class="fa fa-check"></i><b>4.5</b> Video Guide</a></li>
</ul></li>
<li class="part"><span><b>TREE THINKING</b></span></li>
<li class="chapter" data-level="5" data-path="phylogenetics-basics.html"><a href="phylogenetics-basics.html"><i class="fa fa-check"></i><b>5</b> Phylogenetics basics</a>
<ul>
<li class="chapter" data-level="5.1" data-path="phylogenetics-basics.html"><a href="phylogenetics-basics.html#how-to-read-a-phylogenetic-tree"><i class="fa fa-check"></i><b>5.1</b> How to read a phylogenetic tree</a></li>
<li class="chapter" data-level="5.2" data-path="phylogenetics-basics.html"><a href="phylogenetics-basics.html#outgroups"><i class="fa fa-check"></i><b>5.2</b> Outgroups</a></li>
<li class="chapter" data-level="5.3" data-path="phylogenetics-basics.html"><a href="phylogenetics-basics.html#branch-lengths"><i class="fa fa-check"></i><b>5.3</b> Branch lengths</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="visualizing-trees-in-r.html"><a href="visualizing-trees-in-r.html"><i class="fa fa-check"></i><b>6</b> Visualizing trees in R</a>
<ul>
<li class="chapter" data-level="6.1" data-path="visualizing-trees-in-r.html"><a href="visualizing-trees-in-r.html#creating-a-newick-object"><i class="fa fa-check"></i><b>6.1</b> Creating a Newick object</a></li>
<li class="chapter" data-level="6.2" data-path="visualizing-trees-in-r.html"><a href="visualizing-trees-in-r.html#drawing-trees"><i class="fa fa-check"></i><b>6.2</b> Drawing trees</a></li>
<li class="chapter" data-level="6.3" data-path="visualizing-trees-in-r.html"><a href="visualizing-trees-in-r.html#adding-outgroups"><i class="fa fa-check"></i><b>6.3</b> Adding outgroups</a></li>
<li class="chapter" data-level="6.4" data-path="visualizing-trees-in-r.html"><a href="visualizing-trees-in-r.html#drawing-trees-multiple-ways"><i class="fa fa-check"></i><b>6.4</b> Drawing trees multiple ways</a></li>
<li class="chapter" data-level="6.5" data-path="visualizing-trees-in-r.html"><a href="visualizing-trees-in-r.html#the-phylo-class"><i class="fa fa-check"></i><b>6.5</b> The <code>phylo</code> class</a></li>
</ul></li>
<li class="part"><span><b>GETTING SEQUENCES</b></span></li>
<li class="chapter" data-level="7" data-path="what-sequences-should-i-choose.html"><a href="what-sequences-should-i-choose.html"><i class="fa fa-check"></i><b>7</b> What sequences should I choose?</a></li>
<li class="chapter" data-level="8" data-path="finding-sequences-in-genbank.html"><a href="finding-sequences-in-genbank.html"><i class="fa fa-check"></i><b>8</b> Finding sequences in GenBank</a>
<ul>
<li class="chapter" data-level="8.1" data-path="finding-sequences-in-genbank.html"><a href="finding-sequences-in-genbank.html#identifying-a-query-sequence-on-genbank"><i class="fa fa-check"></i><b>8.1</b> Identifying a query sequence on GenBank</a></li>
<li class="chapter" data-level="8.2" data-path="finding-sequences-in-genbank.html"><a href="finding-sequences-in-genbank.html#blastn"><i class="fa fa-check"></i><b>8.2</b> blastn</a></li>
<li class="chapter" data-level="8.3" data-path="finding-sequences-in-genbank.html"><a href="finding-sequences-in-genbank.html#identifying-an-outgroup"><i class="fa fa-check"></i><b>8.3</b> Identifying an outgroup</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="downloading-the-sequences-from-genbank.html"><a href="downloading-the-sequences-from-genbank.html"><i class="fa fa-check"></i><b>9</b> Downloading the sequences from GenBank</a>
<ul>
<li class="chapter" data-level="9.1" data-path="downloading-the-sequences-from-genbank.html"><a href="downloading-the-sequences-from-genbank.html#the-dna.bin-object"><i class="fa fa-check"></i><b>9.1</b> The DNA.bin object</a></li>
<li class="chapter" data-level="9.2" data-path="downloading-the-sequences-from-genbank.html"><a href="downloading-the-sequences-from-genbank.html#the-fasta-format"><i class="fa fa-check"></i><b>9.2</b> The fasta format</a></li>
</ul></li>
<li class="part"><span><b>ALIGNMENT</b></span></li>
<li class="chapter" data-level="10" data-path="why-do-we-align-sequences.html"><a href="why-do-we-align-sequences.html"><i class="fa fa-check"></i><b>10</b> Why do we align sequences?</a></li>
<li class="chapter" data-level="11" data-path="creating-a-multiple-sequence-alignment.html"><a href="creating-a-multiple-sequence-alignment.html"><i class="fa fa-check"></i><b>11</b> Creating a multiple sequence alignment</a>
<ul>
<li class="chapter" data-level="11.1" data-path="creating-a-multiple-sequence-alignment.html"><a href="creating-a-multiple-sequence-alignment.html#installing-the-decipher-package"><i class="fa fa-check"></i><b>11.1</b> Installing the <code>DECIPHER</code> package</a></li>
<li class="chapter" data-level="11.2" data-path="creating-a-multiple-sequence-alignment.html"><a href="creating-a-multiple-sequence-alignment.html#loading-the-fasta-file"><i class="fa fa-check"></i><b>11.2</b> Loading the fasta file</a></li>
<li class="chapter" data-level="11.3" data-path="creating-a-multiple-sequence-alignment.html"><a href="creating-a-multiple-sequence-alignment.html#creating-an-alignment"><i class="fa fa-check"></i><b>11.3</b> Creating an alignment</a></li>
</ul></li>
<li class="part"><span><b>NEIGHBOR JOINING</b></span></li>
<li class="chapter" data-level="12" data-path="the-neighbor-joining-method-of-tree-estimation.html"><a href="the-neighbor-joining-method-of-tree-estimation.html"><i class="fa fa-check"></i><b>12</b> The neighbor joining method of tree estimation</a>
<ul>
<li class="chapter" data-level="12.1" data-path="the-neighbor-joining-method-of-tree-estimation.html"><a href="the-neighbor-joining-method-of-tree-estimation.html#models-of-molecular-evolution"><i class="fa fa-check"></i><b>12.1</b> Models of molecular evolution</a>
<ul>
<li class="chapter" data-level="12.1.1" data-path="the-neighbor-joining-method-of-tree-estimation.html"><a href="the-neighbor-joining-method-of-tree-estimation.html#jc69"><i class="fa fa-check"></i><b>12.1.1</b> JC69</a></li>
<li class="chapter" data-level="12.1.2" data-path="the-neighbor-joining-method-of-tree-estimation.html"><a href="the-neighbor-joining-method-of-tree-estimation.html#k2p"><i class="fa fa-check"></i><b>12.1.2</b> K2P</a></li>
<li class="chapter" data-level="12.1.3" data-path="the-neighbor-joining-method-of-tree-estimation.html"><a href="the-neighbor-joining-method-of-tree-estimation.html#gtr"><i class="fa fa-check"></i><b>12.1.3</b> GTR</a></li>
<li class="chapter" data-level="12.1.4" data-path="the-neighbor-joining-method-of-tree-estimation.html"><a href="the-neighbor-joining-method-of-tree-estimation.html#other-parameters"><i class="fa fa-check"></i><b>12.1.4</b> Other parameters</a></li>
<li class="chapter" data-level="12.1.5" data-path="the-neighbor-joining-method-of-tree-estimation.html"><a href="the-neighbor-joining-method-of-tree-estimation.html#how-do-i-choose-a-model"><i class="fa fa-check"></i><b>12.1.5</b> How do I choose a model?</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="building-a-neighbor-joining-tree.html"><a href="building-a-neighbor-joining-tree.html"><i class="fa fa-check"></i><b>13</b> Building a neighbor joining tree</a>
<ul>
<li class="chapter" data-level="13.1" data-path="building-a-neighbor-joining-tree.html"><a href="building-a-neighbor-joining-tree.html#the-phydat-object"><i class="fa fa-check"></i><b>13.1</b> The phyDat object</a></li>
<li class="chapter" data-level="13.2" data-path="building-a-neighbor-joining-tree.html"><a href="building-a-neighbor-joining-tree.html#choosing-a-substitution-model"><i class="fa fa-check"></i><b>13.2</b> Choosing a substitution model</a></li>
<li class="chapter" data-level="13.3" data-path="building-a-neighbor-joining-tree.html"><a href="building-a-neighbor-joining-tree.html#building-the-neighbor-joining-tree"><i class="fa fa-check"></i><b>13.3</b> Building the neighbor joining tree</a></li>
<li class="chapter" data-level="13.4" data-path="building-a-neighbor-joining-tree.html"><a href="building-a-neighbor-joining-tree.html#visualizing-the-neighbor-joining-tree"><i class="fa fa-check"></i><b>13.4</b> Visualizing the neighbor joining tree</a></li>
<li class="chapter" data-level="13.5" data-path="building-a-neighbor-joining-tree.html"><a href="building-a-neighbor-joining-tree.html#saving-your-trees"><i class="fa fa-check"></i><b>13.5</b> Saving your trees</a></li>
</ul></li>
<li class="part"><span><b>PARSIMONY</b></span></li>
<li class="chapter" data-level="14" data-path="the-parsimony-method-of-tree-estimation.html"><a href="the-parsimony-method-of-tree-estimation.html"><i class="fa fa-check"></i><b>14</b> The parsimony method of tree estimation</a>
<ul>
<li class="chapter" data-level="14.1" data-path="the-parsimony-method-of-tree-estimation.html"><a href="the-parsimony-method-of-tree-estimation.html#parsimony"><i class="fa fa-check"></i><b>14.1</b> Parsimony</a></li>
<li class="chapter" data-level="14.2" data-path="the-parsimony-method-of-tree-estimation.html"><a href="the-parsimony-method-of-tree-estimation.html#tree-searching-methods"><i class="fa fa-check"></i><b>14.2</b> Tree searching methods</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="the-parsimony-method-of-tree-estimation.html"><a href="the-parsimony-method-of-tree-estimation.html#branch-and-bound"><i class="fa fa-check"></i><b>14.2.1</b> Branch and bound</a></li>
<li class="chapter" data-level="14.2.2" data-path="the-parsimony-method-of-tree-estimation.html"><a href="the-parsimony-method-of-tree-estimation.html#nearest-neighbor-interchange-nni"><i class="fa fa-check"></i><b>14.2.2</b> Nearest Neighbor Interchange (NNI)</a></li>
<li class="chapter" data-level="14.2.3" data-path="the-parsimony-method-of-tree-estimation.html"><a href="the-parsimony-method-of-tree-estimation.html#subtree-pruning-and-regrafting-spr"><i class="fa fa-check"></i><b>14.2.3</b> Subtree Pruning and Regrafting (SPR)</a></li>
<li class="chapter" data-level="14.2.4" data-path="the-parsimony-method-of-tree-estimation.html"><a href="the-parsimony-method-of-tree-estimation.html#star-decomposition"><i class="fa fa-check"></i><b>14.2.4</b> Star decomposition</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="building-a-parsimony-tree.html"><a href="building-a-parsimony-tree.html"><i class="fa fa-check"></i><b>15</b> Building a parsimony tree</a>
<ul>
<li class="chapter" data-level="15.1" data-path="building-a-parsimony-tree.html"><a href="building-a-parsimony-tree.html#editing-the-fasta-file"><i class="fa fa-check"></i><b>15.1</b> Editing the fasta file</a></li>
<li class="chapter" data-level="15.2" data-path="building-a-parsimony-tree.html"><a href="building-a-parsimony-tree.html#inferring-a-parsimony-tree"><i class="fa fa-check"></i><b>15.2</b> Inferring a parsimony tree</a></li>
<li class="chapter" data-level="15.3" data-path="building-a-parsimony-tree.html"><a href="building-a-parsimony-tree.html#parsimony-scores"><i class="fa fa-check"></i><b>15.3</b> Parsimony scores</a></li>
<li class="chapter" data-level="15.4" data-path="building-a-parsimony-tree.html"><a href="building-a-parsimony-tree.html#bootstrapping"><i class="fa fa-check"></i><b>15.4</b> Bootstrapping</a></li>
<li class="chapter" data-level="15.5" data-path="building-a-parsimony-tree.html"><a href="building-a-parsimony-tree.html#saving-files"><i class="fa fa-check"></i><b>15.5</b> Saving files</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="about-the-authors.html"><a href="about-the-authors.html"><i class="fa fa-check"></i>About the Authors</a></li>
<li class="chapter" data-level="16" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>16</b> References</a></li>
<li class="divider"></li>
<p style="text-align:center;"> <a href="https://github.com/jhudsl/DaSL_Course_Template_Bookdown" target="blank" > This content was published with</a> <a href="https://bookdown.org/" target="blank"> bookdown by: </a> </p>
<p style="text-align:center;"> <a href="http://jhudatascience.org/"> The Johns Hopkins Data Science Lab </a></p>
<a href="http://jhudatascience.org/"><img src="https://jhudatascience.org/images/dasl.png" style=" width: 80%; padding-left: 40px; padding-top: 8px; vertical-align: top "</a>
<p style="text-align:center; font-size: 12px;"> <a href="https://github.com/rstudio4edu/rstudio4edu-book/"> Style adapted from: rstudio4edu-book </a> <a href ="https://creativecommons.org/licenses/by/2.0/"> (CC-BY 2.0) </a></p>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Molecular Phylogenetic Techniques</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<head>
  <meta name="viewport" content="width=device-width,minimum-scale=1.0,maximum-scale=10.0,initial-scale=1.0">
  <!--script src="https://kit.fontawesome.com/6a26f47516.js"></script-->
  <!--<script src="assets/hideOutput.js"></script>-->
  <link href="assets/AnvIL_style/style.css" rel="stylesheet">
</head>
        


<div class="hero-image-container"> 
  <img class= "hero-image" src="assets/AnVIL_style/dasl_thin_main_image.png">
</div>
<div id="building-a-neighbor-joining-tree" class="section level1" number="13">
<h1><span class="header-section-number">Chapter 13</span> Building a neighbor joining tree</h1>
<p>We are finally ready to start building trees from our data.</p>
<p>For these analyses, we will use an R package called <code>phangorn</code>. If you’d like to learn more about it, you can find the manual <a href="https://cran.r-project.org/web/packages/phangorn/phangorn.pdf">here</a>.</p>
<p><code>phangorn</code> (Phylogenetic Reconstruction and Analysis) is stored in the CRAN repository, so we will use <code>install.packages</code> for the installation.</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="building-a-neighbor-joining-tree.html#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&#39;phangorn&#39;</span>)</span>
<span id="cb49-2"><a href="building-a-neighbor-joining-tree.html#cb49-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-3"><a href="building-a-neighbor-joining-tree.html#cb49-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phangorn)</span></code></pre></div>
<div id="the-phydat-object" class="section level2" number="13.1">
<h2><span class="header-section-number">13.1</span> The phyDat object</h2>
<p><code>phangorn</code> uses a data structure called phyDat to store information. You can either load the fasta file of your alignment into a phyDat object, or you can directly convert the DNAbin object of your alignment into a phyDat object. You will also want to make sure you have access to your DNAbin object, since we will need that for the model testing.</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="building-a-neighbor-joining-tree.html#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Loading a fasta file</span></span>
<span id="cb50-2"><a href="building-a-neighbor-joining-tree.html#cb50-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-3"><a href="building-a-neighbor-joining-tree.html#cb50-3" aria-hidden="true" tabindex="-1"></a>grass.phy <span class="ot">&lt;-</span> <span class="fu">read.phyDat</span>(<span class="st">&#39;grass_aligned.fasta&#39;</span>, <span class="at">format =</span> <span class="st">&#39;fasta&#39;</span>, <span class="at">type =</span> <span class="st">&#39;DNA&#39;</span>)</span>
<span id="cb50-4"><a href="building-a-neighbor-joining-tree.html#cb50-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-5"><a href="building-a-neighbor-joining-tree.html#cb50-5" aria-hidden="true" tabindex="-1"></a><span class="co">#Converting your DNAbin object</span></span>
<span id="cb50-6"><a href="building-a-neighbor-joining-tree.html#cb50-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-7"><a href="building-a-neighbor-joining-tree.html#cb50-7" aria-hidden="true" tabindex="-1"></a>grass.align <span class="ot">&lt;-</span> <span class="fu">read.dna</span>(<span class="st">&#39;grass_aligned.fasta&#39;</span>, <span class="at">format =</span> <span class="st">&#39;fasta&#39;</span>)</span>
<span id="cb50-8"><a href="building-a-neighbor-joining-tree.html#cb50-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-9"><a href="building-a-neighbor-joining-tree.html#cb50-9" aria-hidden="true" tabindex="-1"></a>grass.phy <span class="ot">&lt;-</span> <span class="fu">phyDat</span>(grass.align)</span>
<span id="cb50-10"><a href="building-a-neighbor-joining-tree.html#cb50-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-11"><a href="building-a-neighbor-joining-tree.html#cb50-11" aria-hidden="true" tabindex="-1"></a>grass.phy</span></code></pre></div>
<pre><code>## 13 sequences with 3381 character and 1238 different site patterns.
## The states are a c g t</code></pre>
<p>The phyDat object is remarkably similar to the DNAbin object - this isn’t too surprising, since the same team wrote both <code>ape</code> and <code>phangorn</code>.</p>
<div class="fyi">
<p><strong>R BASICS</strong></p>
<p>If you are pausing your R environment on AnVIL (or have not closed an R session), you may still have all the objects you created previously still available. To check this, you can use the <code>list</code> command.</p>
<p>ls()</p>
<p>R will print all the objects that are still stored in the workspace. If you see your aligned DNAbin object (which can also bee seen in the Environment tab in the upper right box of RStudio), you can directly convert it to a <code>phyDat</code> object instead of loading the fasta file first.</p>
</div>
</div>
<div id="choosing-a-substitution-model" class="section level2" number="13.2">
<h2><span class="header-section-number">13.2</span> Choosing a substitution model</h2>
<p><code>phangorn</code> can test our data against 24 different substitution models to determine which is the best fit. We first have to create a “guide” tree. (In future releases of <code>phangorn</code> creating this tree won’t be necessary, but for now it doesn’t take much extra time to create a tree.)</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="building-a-neighbor-joining-tree.html#cb52-1" aria-hidden="true" tabindex="-1"></a>dist.matrix <span class="ot">&lt;-</span> <span class="fu">dist.dna</span>(grass.align)</span>
<span id="cb52-2"><a href="building-a-neighbor-joining-tree.html#cb52-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-3"><a href="building-a-neighbor-joining-tree.html#cb52-3" aria-hidden="true" tabindex="-1"></a>dist.matrix</span></code></pre></div>
<pre><code>##               JX915632  EF105403.1  DQ073553.1  FJ481575.1  EF204545.1
## EF105403.1 0.018503109                                                
## DQ073553.1 0.018486610 0.009175541                                    
## FJ481575.1 0.053465970 0.059763282 0.062250797                        
## EF204545.1 0.071054126 0.076242240 0.077482792 0.058401695            
## AJ314771.1 0.098387036 0.105024012 0.103575748 0.111726613 0.104200261
## FJ481569.1 0.066921872 0.068194138 0.069428145 0.079685257 0.083719318
## DQ073533.1 0.110258627 0.110258627 0.110174261 0.129491194 0.123022644
## AY804128.1 0.070891254 0.078663432 0.077301449 0.098013117 0.095612138
## AY303125.2 0.134898940 0.140451748 0.140345966 0.156127330 0.155399615
## KF887414.1 0.108883110 0.115798850 0.114317931 0.125235849 0.132797654
## D82941.1   0.229308574 0.234242825 0.230816281 0.245639168 0.252112954
## JX276655.1 0.220197471 0.226712222 0.223309915 0.233015149 0.226285267
##             AJ314771.1  FJ481569.1  DQ073533.1  AY804128.1  AY303125.2
## EF105403.1                                                            
## DQ073553.1                                                            
## FJ481575.1                                                            
## EF204545.1                                                            
## AJ314771.1                                                            
## FJ481569.1 0.116918944                                                
## DQ073533.1 0.157034064 0.105988312                                    
## AY804128.1 0.119801346 0.088583610 0.125139889                        
## AY303125.2 0.175826848 0.145832849 0.125235849 0.156008381            
## KF887414.1 0.147859497 0.107203382 0.121715409 0.133466274 0.139333101
## D82941.1   0.274378224 0.235622139 0.249510345 0.234105767 0.271596554
## JX276655.1 0.254310536 0.229714807 0.252112954 0.233162450 0.258487550
##             KF887414.1    D82941.1
## EF105403.1                        
## DQ073553.1                        
## FJ481575.1                        
## EF204545.1                        
## AJ314771.1                        
## FJ481569.1                        
## DQ073533.1                        
## AY804128.1                        
## AY303125.2                        
## KF887414.1                        
## D82941.1   0.249182286            
## JX276655.1 0.254834070 0.110258627</code></pre>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="building-a-neighbor-joining-tree.html#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(dist.matrix)</span></code></pre></div>
<pre><code>##  &#39;dist&#39; num [1:78] 0.0185 0.0185 0.0535 0.0711 0.0984 ...
##  - attr(*, &quot;Size&quot;)= int 13
##  - attr(*, &quot;Labels&quot;)= chr [1:13] &quot;JX915632&quot; &quot;EF105403.1&quot; &quot;DQ073553.1&quot; &quot;FJ481575.1&quot; ...
##  - attr(*, &quot;Upper&quot;)= logi FALSE
##  - attr(*, &quot;Diag&quot;)= logi FALSE
##  - attr(*, &quot;call&quot;)= language dist.dna(x = grass.align)
##  - attr(*, &quot;method&quot;)= chr &quot;K80&quot;</code></pre>
<p>The distance matrix stores all the distances in the lower half of the table to save space. This particular matrix used the K80 model (also known as the K2P model) to calculate the distances. When we use the <code>str(dist.matrix)</code> command, we can see the model used stored in the method attribute.</p>
<p>Next, we make a very quick neighbor joining tree to act as the guide tree. We then run the model test command using our phyDat object and the guide tree.</p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="building-a-neighbor-joining-tree.html#cb56-1" aria-hidden="true" tabindex="-1"></a>tree.guide <span class="ot">&lt;-</span> <span class="fu">NJ</span>(dist.matrix)</span>
<span id="cb56-2"><a href="building-a-neighbor-joining-tree.html#cb56-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-3"><a href="building-a-neighbor-joining-tree.html#cb56-3" aria-hidden="true" tabindex="-1"></a>mod.test <span class="ot">&lt;-</span> <span class="fu">modelTest</span>(grass.phy, <span class="at">tree =</span> tree.guide)</span></code></pre></div>
<pre><code>## [1] &quot;JC+I&quot;
## [1] &quot;JC+G&quot;
## [1] &quot;JC+G+I&quot;
## [1] &quot;F81+I&quot;
## [1] &quot;F81+G&quot;
## [1] &quot;F81+G+I&quot;
## [1] &quot;K80+I&quot;
## [1] &quot;K80+G&quot;
## [1] &quot;K80+G+I&quot;
## [1] &quot;HKY+I&quot;
## [1] &quot;HKY+G&quot;
## [1] &quot;HKY+G+I&quot;
## [1] &quot;SYM+I&quot;
## [1] &quot;SYM+G&quot;
## [1] &quot;SYM+G+I&quot;
## [1] &quot;GTR+I&quot;
## [1] &quot;GTR+G&quot;
## [1] &quot;GTR+G+I&quot;</code></pre>
<p>As the model test runs, R prints out most of the models being tested. For some reason, R doesn’t print the basic models. The first three models listed are JC + I, JC + G, and JC + G + I. While these models are being tested, the program is <em>also</em> testing the basic JC model.</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="building-a-neighbor-joining-tree.html#cb58-1" aria-hidden="true" tabindex="-1"></a>mod.test</span></code></pre></div>
<pre><code>##      Model df    logLik      AIC          AICw     AICc         AICcw      BIC
## 1       JC 23 -12889.65 25825.30 2.700386e-211 25825.63 3.157131e-211 25966.20
## 2     JC+I 24 -12848.73 25745.47 5.846250e-194 25745.82 6.737694e-194 25892.49
## 3     JC+G 24 -12841.04 25730.08 1.284266e-190 25730.43 1.480093e-190 25877.10
## 4   JC+G+I 25 -12841.04 25732.08 4.724094e-191 25732.46 5.363607e-191 25885.23
## 5      F81 26 -12712.05 25476.11 1.807454e-135 25476.53 2.020447e-135 25635.38
## 6    F81+I 27 -12670.12 25394.24 1.083122e-117 25394.69 1.191340e-117 25559.64
## 7    F81+G 27 -12661.52 25377.04 5.877060e-114 25377.49 6.464260e-114 25542.44
## 8  F81+G+I 28 -12661.52 25379.04 2.161804e-114 25379.53 2.338246e-114 25550.57
## 9      K80 24 -12675.91 25399.82 6.659747e-119 25400.18 7.675234e-119 25546.84
## 10   K80+I 25 -12625.77 25301.54  1.460748e-97 25301.93  1.658493e-97 25454.69
## 11   K80+G 25 -12615.42 25280.84  4.559902e-93 25281.23  5.177187e-93 25433.99
## 12 K80+G+I 26 -12615.42 25282.84  1.677274e-93 25283.26  1.874926e-93 25442.12
## 13     HKY 27 -12490.04 25034.08  1.748144e-39 25034.53  1.922808e-39 25199.48
## 14   HKY+I 28 -12441.45 24938.91  8.118092e-19 24939.39  8.780675e-19 25110.43
## 15   HKY+G 28 -12430.70 24917.40  3.787695e-14 24917.89  4.096840e-14 25088.93
## 16 HKY+G+I 29 -12430.70 24919.40  1.393244e-14 24919.92  1.480992e-14 25097.06
## 17     SYM 28 -12666.82 25389.65 1.076040e-116 25390.13 1.163864e-116 25561.17
## 18   SYM+I 29 -12619.36 25296.72  1.621861e-96 25297.24  1.724008e-96 25474.38
## 19   SYM+G 29 -12609.63 25277.27  2.724038e-92 25277.79  2.895602e-92 25454.92
## 20 SYM+G+I 30 -12609.63 25279.27  1.001914e-92 25279.82  1.046030e-92 25463.04
## 21     GTR 31 -12462.58 24987.15  2.709162e-29 24987.74  2.776339e-29 25177.06
## 22   GTR+I 32 -12406.95 24877.89  1.439643e-05 24878.52  1.447278e-05 25073.92
## 23   GTR+G 32 -12396.11 24856.22  7.310833e-01 24856.85  7.349605e-01 25052.25
## 24 GTR+G+I 33 -12396.11 24858.22  2.689024e-01 24858.89  2.650250e-01 25060.38</code></pre>
<p><code>phangorn</code> calculates multiple statistics that can be used to judge the fit of all the models. Which one you choose depends on what you prefer, although the field standard is generally to use the Akaike Information Criterion. This is a single number that combines how well the model fits the data (as determined by the log likelihood score) with a penalty for increasing model complexity. More complex models will almost always fit the data better, but in small datasets it can be difficult to get good estimates of every parameter when the model is complex. With AIC, we might find the GTR + G + I model fits the best, but we only have enough samples to estimates 2 parameters with any confidence. In this case, the AIC score might tell us to use the K80 (K2P) model.</p>
<p>Smaller numbers are better when it comes to AIC. This is also the case for AICc (which is a “second generation” or updated calculate of AIC) and BIC (the Bayesian Information Criterion, a version of AIC that includes a stronger penalty for additional parameters). In RStudio, you canclick on “mod.test” in the Environment tab from the pane in the upper right corner. That will open the results of our model test analysis in a separate window. If we click on “AIC” or “AICc”, we can order the models by the AIC values. When we do this, we see the lowest AIC and AICc values are for the GTR + G model, which is what we will use going forward.</p>
</div>
<div id="building-the-neighbor-joining-tree" class="section level2" number="13.3">
<h2><span class="header-section-number">13.3</span> Building the neighbor joining tree</h2>
<p>We’ve already calculated a distance matrix above, but this time we will specify the model we want to use.</p>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="building-a-neighbor-joining-tree.html#cb60-1" aria-hidden="true" tabindex="-1"></a>dist.matrix <span class="ot">&lt;-</span> <span class="fu">dist.dna</span>(grass.align, <span class="at">model =</span> <span class="st">&quot;GTR + G&quot;</span>)</span>
<span id="cb60-2"><a href="building-a-neighbor-joining-tree.html#cb60-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-3"><a href="building-a-neighbor-joining-tree.html#cb60-3" aria-hidden="true" tabindex="-1"></a>tree <span class="ot">&lt;-</span> <span class="fu">NJ</span>(dist.matrix)</span>
<span id="cb60-4"><a href="building-a-neighbor-joining-tree.html#cb60-4" aria-hidden="true" tabindex="-1"></a>tree</span></code></pre></div>
<p>But wait, why are we getting an error message? As it turns out, the <code>modelTest</code> command might test 24 different models, but the <code>dist.dna</code> command can’t use all of those models. This is frustrating but also not uncommon when doing phylogenetic analyses. Unfortunately, not every program supports every substitution model.</p>
<p><code>dist.dna</code> appears to support the following models: RAW, JC69, K80, F81, K81, F84, T92, TN93, GG95, LOGDET, BH87, PARALIN, N, TS, TV, INDEL, INDELBLOCK. We can check which of these models has the lowest AIC from our earlier model test. Surprisingly, it’s the K80 model, which is what we used initially.</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="building-a-neighbor-joining-tree.html#cb61-1" aria-hidden="true" tabindex="-1"></a>dist.matrix <span class="ot">&lt;-</span> <span class="fu">dist.dna</span>(grass.align, <span class="at">model =</span> <span class="st">&quot;K80&quot;</span>)</span>
<span id="cb61-2"><a href="building-a-neighbor-joining-tree.html#cb61-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-3"><a href="building-a-neighbor-joining-tree.html#cb61-3" aria-hidden="true" tabindex="-1"></a>tree <span class="ot">&lt;-</span> <span class="fu">NJ</span>(dist.matrix)</span></code></pre></div>
</div>
<div id="visualizing-the-neighbor-joining-tree" class="section level2" number="13.4">
<h2><span class="header-section-number">13.4</span> Visualizing the neighbor joining tree</h2>
<p>Let’s take a look at the neighbor joining tree.</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="building-a-neighbor-joining-tree.html#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(tree, <span class="at">type =</span> <span class="st">&quot;unrooted&quot;</span>)</span></code></pre></div>
<p><img src="resources/images/06-neighbor-joining_files/figure-html/unnamed-chunk-11-1.png" width="672" /></p>
<p>While we have indeed generated a tree, it’s not really the easiest to interpret at the moment. It’s hard to see what the relationships among taxa are because the tips are labeled with GenBank accession numbers instead of taxa names. Also, we really ought to declare an outgroup.</p>
<p>First let’s change the tip labels. This is simply a matter of replacing each accession number with the taxon name. We do this by creating a vector of taxon names (matching the order of the accession numbers), then replacing the <code>tip.label</code> variable in our <code>tree</code> object. Because most of us are not plant experts, we’ll use the common names for each sample, but it’s also acceptable to use the scientific names.</p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="building-a-neighbor-joining-tree.html#cb63-1" aria-hidden="true" tabindex="-1"></a>tree<span class="sc">$</span>tip.label</span></code></pre></div>
<pre><code>##  [1] &quot;JX915632&quot;   &quot;EF105403.1&quot; &quot;DQ073553.1&quot; &quot;FJ481575.1&quot; &quot;EF204545.1&quot;
##  [6] &quot;AJ314771.1&quot; &quot;FJ481569.1&quot; &quot;DQ073533.1&quot; &quot;AY804128.1&quot; &quot;AY303125.2&quot;
## [11] &quot;KF887414.1&quot; &quot;D82941.1&quot;   &quot;JX276655.1&quot;</code></pre>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="building-a-neighbor-joining-tree.html#cb65-1" aria-hidden="true" tabindex="-1"></a>new.labels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;wheat&#39;</span>, <span class="st">&#39;intermediate wheatgrass&#39;</span>, <span class="st">&#39;mammoth wild rye&#39;</span>, <span class="st">&#39;wheatgrass&#39;</span>, <span class="st">&#39;tall wheatgrass&#39;</span>, <span class="st">&#39;rye&#39;</span>, <span class="st">&#39;Asiatic grass&#39;</span>, <span class="st">&#39;crested wheatgrass&#39;</span>, <span class="st">&#39;Tauschs goatgrass&#39;</span>, <span class="st">&#39;medusahead rye&#39;</span>, <span class="st">&#39;mosquito grass&#39;</span>, <span class="st">&#39;barley_D-hordein&#39;</span>, <span class="st">&#39;Siberian wild rye_D-hordein&#39;</span> )</span>
<span id="cb65-2"><a href="building-a-neighbor-joining-tree.html#cb65-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb65-3"><a href="building-a-neighbor-joining-tree.html#cb65-3" aria-hidden="true" tabindex="-1"></a>tree<span class="sc">$</span>tip.label <span class="ot">&lt;-</span> new.labels</span></code></pre></div>
<p>Next, we declare our outgroup (in this case, the two D-hordein samples) and root our tree. (Remember, we did this in our first exercise using R.)</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="building-a-neighbor-joining-tree.html#cb66-1" aria-hidden="true" tabindex="-1"></a>tree.root <span class="ot">&lt;-</span> <span class="fu">root</span>(tree, <span class="at">outgroup =</span> <span class="fu">c</span>(<span class="st">&#39;barley_D-hordein&#39;</span>,<span class="st">&#39;Siberian wild rye_D-hordein&#39;</span>))</span>
<span id="cb66-2"><a href="building-a-neighbor-joining-tree.html#cb66-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-3"><a href="building-a-neighbor-joining-tree.html#cb66-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(tree.root, <span class="at">type =</span> <span class="st">&quot;phylogram&quot;</span>, <span class="at">main =</span> <span class="st">&#39;Neighbor Joining, Glu-1&#39;</span>)</span></code></pre></div>
<p><img src="resources/images/06-neighbor-joining_files/figure-html/unnamed-chunk-13-1.png" width="672" /></p>
<p>Now we have a tree that we can begin to make sense of. To an non-botanist, it seems interesting the samples with the common name “wheatgrass” don’t appear to be clustering together. Common names can be misleading about phylogenetic relationships!</p>
</div>
<div id="saving-your-trees" class="section level2" number="13.5">
<h2><span class="header-section-number">13.5</span> Saving your trees</h2>
<p>You want to be sure to save the rooted tree and model test results to the persistent disk.</p>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="building-a-neighbor-joining-tree.html#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.tree</span>(tree.root, <span class="at">file =</span> <span class="st">&#39;nj_grass.tre&#39;</span>)</span>
<span id="cb67-2"><a href="building-a-neighbor-joining-tree.html#cb67-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(mod.test, <span class="at">file =</span> <span class="st">&#39;grass_model_test&#39;</span>, <span class="at">quote=</span>F, <span class="at">sep=</span><span class="st">&#39;</span><span class="sc">\t</span><span class="st">&#39;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="building-a-neighbor-joining-tree.html#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.0.2 (2020-06-22)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.3 LTS
## 
## Matrix products: default
## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats4    parallel  stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
## [1] phangorn_2.5.5      DECIPHER_2.18.1     RSQLite_2.2.1      
## [4] Biostrings_2.58.0   XVector_0.30.0      IRanges_2.24.1     
## [7] S4Vectors_0.28.1    BiocGenerics_0.36.1 ape_5.4-1          
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_1.0.8      highr_0.8       pillar_1.4.6    compiler_4.0.2 
##  [5] jquerylib_0.1.4 tools_4.0.2     zlibbioc_1.36.0 bit_4.0.4      
##  [9] digest_0.6.25   memoise_1.1.0   evaluate_0.14   lifecycle_1.0.0
## [13] tibble_3.0.3    nlme_3.1-149    lattice_0.20-41 pkgconfig_2.0.3
## [17] rlang_0.4.10    igraph_1.2.6    fastmatch_1.1-0 Matrix_1.2-18  
## [21] DBI_1.1.0       yaml_2.2.1      xfun_0.26       stringr_1.4.0  
## [25] knitr_1.33      fs_1.5.0        vctrs_0.3.4     hms_0.5.3      
## [29] bit64_4.0.5     grid_4.0.2      R6_2.4.1        ottrpal_0.1.2  
## [33] rmarkdown_2.10  bookdown_0.24   blob_1.2.1      readr_1.4.0    
## [37] magrittr_2.0.2  ellipsis_0.3.1  htmltools_0.5.0 quadprog_1.5-8 
## [41] stringi_1.5.3   crayon_1.3.4</code></pre>

</div>
</div>



<hr>
<center> 
  <div class="footer">
      All illustrations <a href="https://creativecommons.org/licenses/by/4.0/">CC-BY. </a>
      <br>
      All other materials <a href= "https://creativecommons.org/licenses/by/4.0/"> CC-BY </a> unless noted otherwise.
  </div>
</center>
            </section>

          </div>
        </div>
      </div>
<a href="the-neighbor-joining-method-of-tree-estimation.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="the-parsimony-method-of-tree-estimation.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
