<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 9 Downloading the sequences from GenBank | Molecular Phylogenetic Techniques</title>
  <meta name="description" content="Description about Course/Book." />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 9 Downloading the sequences from GenBank | Molecular Phylogenetic Techniques" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Description about Course/Book." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 9 Downloading the sequences from GenBank | Molecular Phylogenetic Techniques" />
  
  <meta name="twitter:description" content="Description about Course/Book." />
  



<meta name="date" content="2022-08-26" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="assets/AnVIL_style/anvil_favicon.ico" type="image/x-icon" />
<link rel="prev" href="finding-sequences-in-genbank.html"/>
<link rel="next" href="why-do-we-align-sequences.html"/>
<script src="libs/header-attrs-2.10/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="assets/AnVIL_style/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<a href="https://anvilproject.org/" target="_blank"><img src="assets/AnVIL_style/logo-anvil.png" style="width: 80%; padding-left: 15px; padding-top: 8px;"</a>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>About this Book</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#anvil-collection"><i class="fa fa-check"></i>AnVIL Collection</a></li>
</ul></li>
<li class="part"><span><b>BOOK OVERVIEW</b></span></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#motivation"><i class="fa fa-check"></i><b>1.1</b> Motivation</a></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#target-audience"><i class="fa fa-check"></i><b>1.2</b> Target Audience</a></li>
<li class="chapter" data-level="1.3" data-path="introduction.html"><a href="introduction.html#curriculum"><i class="fa fa-check"></i><b>1.3</b> Curriculum</a></li>
</ul></li>
<li class="part"><span><b>BEFORE YOU START</b></span></li>
<li class="chapter" data-level="2" data-path="sign-in.html"><a href="sign-in.html"><i class="fa fa-check"></i><b>2</b> Sign in</a></li>
<li class="chapter" data-level="3" data-path="create-a-workspace.html"><a href="create-a-workspace.html"><i class="fa fa-check"></i><b>3</b> Create a workspace</a></li>
<li class="part"><span><b>RUNNING RSTUDIO</b></span></li>
<li class="chapter" data-level="4" data-path="rstudio.html"><a href="rstudio.html"><i class="fa fa-check"></i><b>4</b> Working in RStudio</a>
<ul>
<li class="chapter" data-level="4.1" data-path="rstudio.html"><a href="rstudio.html#launch-rstudio-cloud-environment"><i class="fa fa-check"></i><b>4.1</b> Launch RStudio Cloud Environment</a></li>
<li class="chapter" data-level="4.2" data-path="rstudio.html"><a href="rstudio.html#tour-rstudio"><i class="fa fa-check"></i><b>4.2</b> Tour RStudio</a></li>
<li class="chapter" data-level="4.3" data-path="rstudio.html"><a href="rstudio.html#stopping"><i class="fa fa-check"></i><b>4.3</b> Pause RStudio</a></li>
<li class="chapter" data-level="4.4" data-path="rstudio.html"><a href="rstudio.html#delete-rstudio-cloud-environment"><i class="fa fa-check"></i><b>4.4</b> Delete RStudio Cloud Environment</a></li>
<li class="chapter" data-level="4.5" data-path="rstudio.html"><a href="rstudio.html#video-guide"><i class="fa fa-check"></i><b>4.5</b> Video Guide</a></li>
</ul></li>
<li class="part"><span><b>TREE THINKING</b></span></li>
<li class="chapter" data-level="5" data-path="phylogenetics-basics.html"><a href="phylogenetics-basics.html"><i class="fa fa-check"></i><b>5</b> Phylogenetics basics</a>
<ul>
<li class="chapter" data-level="5.1" data-path="phylogenetics-basics.html"><a href="phylogenetics-basics.html#how-to-read-a-phylogenetic-tree"><i class="fa fa-check"></i><b>5.1</b> How to read a phylogenetic tree</a></li>
<li class="chapter" data-level="5.2" data-path="phylogenetics-basics.html"><a href="phylogenetics-basics.html#outgroups"><i class="fa fa-check"></i><b>5.2</b> Outgroups</a></li>
<li class="chapter" data-level="5.3" data-path="phylogenetics-basics.html"><a href="phylogenetics-basics.html#branch-lengths"><i class="fa fa-check"></i><b>5.3</b> Branch lengths</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="visualizing-trees-in-r.html"><a href="visualizing-trees-in-r.html"><i class="fa fa-check"></i><b>6</b> Visualizing trees in R</a>
<ul>
<li class="chapter" data-level="6.1" data-path="visualizing-trees-in-r.html"><a href="visualizing-trees-in-r.html#creating-a-newick-object"><i class="fa fa-check"></i><b>6.1</b> Creating a Newick object</a></li>
<li class="chapter" data-level="6.2" data-path="visualizing-trees-in-r.html"><a href="visualizing-trees-in-r.html#drawing-trees"><i class="fa fa-check"></i><b>6.2</b> Drawing trees</a></li>
<li class="chapter" data-level="6.3" data-path="visualizing-trees-in-r.html"><a href="visualizing-trees-in-r.html#adding-outgroups"><i class="fa fa-check"></i><b>6.3</b> Adding outgroups</a></li>
<li class="chapter" data-level="6.4" data-path="visualizing-trees-in-r.html"><a href="visualizing-trees-in-r.html#drawing-trees-multiple-ways"><i class="fa fa-check"></i><b>6.4</b> Drawing trees multiple ways</a></li>
<li class="chapter" data-level="6.5" data-path="visualizing-trees-in-r.html"><a href="visualizing-trees-in-r.html#the-phylo-class"><i class="fa fa-check"></i><b>6.5</b> The <code>phylo</code> class</a></li>
</ul></li>
<li class="part"><span><b>GETTING SEQUENCES</b></span></li>
<li class="chapter" data-level="7" data-path="what-sequences-should-i-choose.html"><a href="what-sequences-should-i-choose.html"><i class="fa fa-check"></i><b>7</b> What sequences should I choose?</a></li>
<li class="chapter" data-level="8" data-path="finding-sequences-in-genbank.html"><a href="finding-sequences-in-genbank.html"><i class="fa fa-check"></i><b>8</b> Finding sequences in GenBank</a>
<ul>
<li class="chapter" data-level="8.1" data-path="finding-sequences-in-genbank.html"><a href="finding-sequences-in-genbank.html#identifying-a-query-sequence-on-genbank"><i class="fa fa-check"></i><b>8.1</b> Identifying a query sequence on GenBank</a></li>
<li class="chapter" data-level="8.2" data-path="finding-sequences-in-genbank.html"><a href="finding-sequences-in-genbank.html#blastn"><i class="fa fa-check"></i><b>8.2</b> blastn</a></li>
<li class="chapter" data-level="8.3" data-path="finding-sequences-in-genbank.html"><a href="finding-sequences-in-genbank.html#identifying-an-outgroup"><i class="fa fa-check"></i><b>8.3</b> Identifying an outgroup</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="downloading-the-sequences-from-genbank.html"><a href="downloading-the-sequences-from-genbank.html"><i class="fa fa-check"></i><b>9</b> Downloading the sequences from GenBank</a>
<ul>
<li class="chapter" data-level="9.1" data-path="downloading-the-sequences-from-genbank.html"><a href="downloading-the-sequences-from-genbank.html#the-dna.bin-object"><i class="fa fa-check"></i><b>9.1</b> The DNA.bin object</a></li>
<li class="chapter" data-level="9.2" data-path="downloading-the-sequences-from-genbank.html"><a href="downloading-the-sequences-from-genbank.html#the-fasta-format"><i class="fa fa-check"></i><b>9.2</b> The fasta format</a></li>
</ul></li>
<li class="part"><span><b>ALIGNMENT</b></span></li>
<li class="chapter" data-level="10" data-path="why-do-we-align-sequences.html"><a href="why-do-we-align-sequences.html"><i class="fa fa-check"></i><b>10</b> Why do we align sequences?</a></li>
<li class="chapter" data-level="11" data-path="creating-a-multiple-sequence-alignment.html"><a href="creating-a-multiple-sequence-alignment.html"><i class="fa fa-check"></i><b>11</b> Creating a multiple sequence alignment</a>
<ul>
<li class="chapter" data-level="11.1" data-path="creating-a-multiple-sequence-alignment.html"><a href="creating-a-multiple-sequence-alignment.html#installing-the-decipher-package"><i class="fa fa-check"></i><b>11.1</b> Installing the <code>DECIPHER</code> package</a></li>
<li class="chapter" data-level="11.2" data-path="creating-a-multiple-sequence-alignment.html"><a href="creating-a-multiple-sequence-alignment.html#loading-the-fasta-file"><i class="fa fa-check"></i><b>11.2</b> Loading the fasta file</a></li>
<li class="chapter" data-level="11.3" data-path="creating-a-multiple-sequence-alignment.html"><a href="creating-a-multiple-sequence-alignment.html#creating-an-alignment"><i class="fa fa-check"></i><b>11.3</b> Creating an alignment</a></li>
</ul></li>
<li class="part"><span><b>NEIGHBOR JOINING</b></span></li>
<li class="chapter" data-level="12" data-path="the-neighbor-joining-method-of-tree-estimation.html"><a href="the-neighbor-joining-method-of-tree-estimation.html"><i class="fa fa-check"></i><b>12</b> The neighbor joining method of tree estimation</a>
<ul>
<li class="chapter" data-level="12.1" data-path="the-neighbor-joining-method-of-tree-estimation.html"><a href="the-neighbor-joining-method-of-tree-estimation.html#models-of-molecular-evolution"><i class="fa fa-check"></i><b>12.1</b> Models of molecular evolution</a>
<ul>
<li class="chapter" data-level="12.1.1" data-path="the-neighbor-joining-method-of-tree-estimation.html"><a href="the-neighbor-joining-method-of-tree-estimation.html#jc69"><i class="fa fa-check"></i><b>12.1.1</b> JC69</a></li>
<li class="chapter" data-level="12.1.2" data-path="the-neighbor-joining-method-of-tree-estimation.html"><a href="the-neighbor-joining-method-of-tree-estimation.html#k2p"><i class="fa fa-check"></i><b>12.1.2</b> K2P</a></li>
<li class="chapter" data-level="12.1.3" data-path="the-neighbor-joining-method-of-tree-estimation.html"><a href="the-neighbor-joining-method-of-tree-estimation.html#gtr"><i class="fa fa-check"></i><b>12.1.3</b> GTR</a></li>
<li class="chapter" data-level="12.1.4" data-path="the-neighbor-joining-method-of-tree-estimation.html"><a href="the-neighbor-joining-method-of-tree-estimation.html#other-parameters"><i class="fa fa-check"></i><b>12.1.4</b> Other parameters</a></li>
<li class="chapter" data-level="12.1.5" data-path="the-neighbor-joining-method-of-tree-estimation.html"><a href="the-neighbor-joining-method-of-tree-estimation.html#how-do-i-choose-a-model"><i class="fa fa-check"></i><b>12.1.5</b> How do I choose a model?</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="building-a-neighbor-joining-tree.html"><a href="building-a-neighbor-joining-tree.html"><i class="fa fa-check"></i><b>13</b> Building a neighbor joining tree</a>
<ul>
<li class="chapter" data-level="13.1" data-path="building-a-neighbor-joining-tree.html"><a href="building-a-neighbor-joining-tree.html#the-phydat-object"><i class="fa fa-check"></i><b>13.1</b> The phyDat object</a></li>
<li class="chapter" data-level="13.2" data-path="building-a-neighbor-joining-tree.html"><a href="building-a-neighbor-joining-tree.html#choosing-a-substitution-model"><i class="fa fa-check"></i><b>13.2</b> Choosing a substitution model</a></li>
<li class="chapter" data-level="13.3" data-path="building-a-neighbor-joining-tree.html"><a href="building-a-neighbor-joining-tree.html#building-the-neighbor-joining-tree"><i class="fa fa-check"></i><b>13.3</b> Building the neighbor joining tree</a></li>
<li class="chapter" data-level="13.4" data-path="building-a-neighbor-joining-tree.html"><a href="building-a-neighbor-joining-tree.html#visualizing-the-neighbor-joining-tree"><i class="fa fa-check"></i><b>13.4</b> Visualizing the neighbor joining tree</a></li>
<li class="chapter" data-level="13.5" data-path="building-a-neighbor-joining-tree.html"><a href="building-a-neighbor-joining-tree.html#saving-your-trees"><i class="fa fa-check"></i><b>13.5</b> Saving your trees</a></li>
</ul></li>
<li class="part"><span><b>PARSIMONY</b></span></li>
<li class="chapter" data-level="14" data-path="the-parsimony-method-of-tree-estimation.html"><a href="the-parsimony-method-of-tree-estimation.html"><i class="fa fa-check"></i><b>14</b> The parsimony method of tree estimation</a>
<ul>
<li class="chapter" data-level="14.1" data-path="the-parsimony-method-of-tree-estimation.html"><a href="the-parsimony-method-of-tree-estimation.html#parsimony"><i class="fa fa-check"></i><b>14.1</b> Parsimony</a></li>
<li class="chapter" data-level="14.2" data-path="the-parsimony-method-of-tree-estimation.html"><a href="the-parsimony-method-of-tree-estimation.html#tree-searching-methods"><i class="fa fa-check"></i><b>14.2</b> Tree searching methods</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="the-parsimony-method-of-tree-estimation.html"><a href="the-parsimony-method-of-tree-estimation.html#branch-and-bound"><i class="fa fa-check"></i><b>14.2.1</b> Branch and bound</a></li>
<li class="chapter" data-level="14.2.2" data-path="the-parsimony-method-of-tree-estimation.html"><a href="the-parsimony-method-of-tree-estimation.html#nearest-neighbor-interchange-nni"><i class="fa fa-check"></i><b>14.2.2</b> Nearest Neighbor Interchange (NNI)</a></li>
<li class="chapter" data-level="14.2.3" data-path="the-parsimony-method-of-tree-estimation.html"><a href="the-parsimony-method-of-tree-estimation.html#subtree-pruning-and-regrafting-spr"><i class="fa fa-check"></i><b>14.2.3</b> Subtree Pruning and Regrafting (SPR)</a></li>
<li class="chapter" data-level="14.2.4" data-path="the-parsimony-method-of-tree-estimation.html"><a href="the-parsimony-method-of-tree-estimation.html#star-decomposition"><i class="fa fa-check"></i><b>14.2.4</b> Star decomposition</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="building-a-parsimony-tree.html"><a href="building-a-parsimony-tree.html"><i class="fa fa-check"></i><b>15</b> Building a parsimony tree</a>
<ul>
<li class="chapter" data-level="15.1" data-path="building-a-parsimony-tree.html"><a href="building-a-parsimony-tree.html#editing-the-fasta-file"><i class="fa fa-check"></i><b>15.1</b> Editing the fasta file</a></li>
<li class="chapter" data-level="15.2" data-path="building-a-parsimony-tree.html"><a href="building-a-parsimony-tree.html#inferring-a-parsimony-tree"><i class="fa fa-check"></i><b>15.2</b> Inferring a parsimony tree</a></li>
<li class="chapter" data-level="15.3" data-path="building-a-parsimony-tree.html"><a href="building-a-parsimony-tree.html#parsimony-scores"><i class="fa fa-check"></i><b>15.3</b> Parsimony scores</a></li>
<li class="chapter" data-level="15.4" data-path="building-a-parsimony-tree.html"><a href="building-a-parsimony-tree.html#bootstrapping"><i class="fa fa-check"></i><b>15.4</b> Bootstrapping</a></li>
<li class="chapter" data-level="15.5" data-path="building-a-parsimony-tree.html"><a href="building-a-parsimony-tree.html#saving-files"><i class="fa fa-check"></i><b>15.5</b> Saving files</a></li>
</ul></li>
<li class="part"><span><b>MORE TREE THINKING</b></span></li>
<li class="chapter" data-level="16" data-path="more-phylogenetics-basics.html"><a href="more-phylogenetics-basics.html"><i class="fa fa-check"></i><b>16</b> More Phylogenetics Basics</a>
<ul>
<li class="chapter" data-level="16.1" data-path="more-phylogenetics-basics.html"><a href="more-phylogenetics-basics.html#tree-topology"><i class="fa fa-check"></i><b>16.1</b> Tree topology</a></li>
<li class="chapter" data-level="16.2" data-path="more-phylogenetics-basics.html"><a href="more-phylogenetics-basics.html#outgroups-1"><i class="fa fa-check"></i><b>16.2</b> Outgroups</a></li>
<li class="chapter" data-level="16.3" data-path="more-phylogenetics-basics.html"><a href="more-phylogenetics-basics.html#comparing-trees"><i class="fa fa-check"></i><b>16.3</b> Comparing trees</a></li>
</ul></li>
<li class="part"><span><b>MAXIMUM LIKELIHOOD</b></span></li>
<li class="chapter" data-level="17" data-path="what-is-maximum-likelihood.html"><a href="what-is-maximum-likelihood.html"><i class="fa fa-check"></i><b>17</b> What is Maximum Likelihood?</a></li>
<li class="chapter" data-level="18" data-path="estimating-a-tree-using-maximum-likelihood.html"><a href="estimating-a-tree-using-maximum-likelihood.html"><i class="fa fa-check"></i><b>18</b> Estimating a tree using Maximum Likelihood</a>
<ul>
<li class="chapter" data-level="18.1" data-path="estimating-a-tree-using-maximum-likelihood.html"><a href="estimating-a-tree-using-maximum-likelihood.html#creating-a-starting-tree"><i class="fa fa-check"></i><b>18.1</b> Creating a starting tree</a></li>
<li class="chapter" data-level="18.2" data-path="estimating-a-tree-using-maximum-likelihood.html"><a href="estimating-a-tree-using-maximum-likelihood.html#calculating-likelihood-score"><i class="fa fa-check"></i><b>18.2</b> Calculating likelihood score</a></li>
<li class="chapter" data-level="18.3" data-path="estimating-a-tree-using-maximum-likelihood.html"><a href="estimating-a-tree-using-maximum-likelihood.html#optimizing-the-maximum-likelihood-parameters"><i class="fa fa-check"></i><b>18.3</b> Optimizing the maximum likelihood parameters</a></li>
<li class="chapter" data-level="18.4" data-path="estimating-a-tree-using-maximum-likelihood.html"><a href="estimating-a-tree-using-maximum-likelihood.html#changing-the-model-for-your-ml-search"><i class="fa fa-check"></i><b>18.4</b> Changing the model for your ML search</a></li>
<li class="chapter" data-level="18.5" data-path="estimating-a-tree-using-maximum-likelihood.html"><a href="estimating-a-tree-using-maximum-likelihood.html#bootstrapping-the-ml-tree"><i class="fa fa-check"></i><b>18.5</b> Bootstrapping the ML tree</a></li>
<li class="chapter" data-level="18.6" data-path="estimating-a-tree-using-maximum-likelihood.html"><a href="estimating-a-tree-using-maximum-likelihood.html#saving-your-bootstrap-tree"><i class="fa fa-check"></i><b>18.6</b> Saving your bootstrap tree</a></li>
</ul></li>
<li class="part"><span><b>TIME TREES</b></span></li>
<li class="chapter" data-level="19" data-path="what-is-a-time-tree.html"><a href="what-is-a-time-tree.html"><i class="fa fa-check"></i><b>19</b> What is a time tree?</a>
<ul>
<li class="chapter" data-level="19.1" data-path="what-is-a-time-tree.html"><a href="what-is-a-time-tree.html#molecular-clocks"><i class="fa fa-check"></i><b>19.1</b> Molecular clocks</a></li>
<li class="chapter" data-level="19.2" data-path="what-is-a-time-tree.html"><a href="what-is-a-time-tree.html#calibration-points"><i class="fa fa-check"></i><b>19.2</b> Calibration points</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="adding-calibration-points-to-your-phylogenies.html"><a href="adding-calibration-points-to-your-phylogenies.html"><i class="fa fa-check"></i><b>20</b> Adding calibration points to your phylogenies</a>
<ul>
<li class="chapter" data-level="20.1" data-path="adding-calibration-points-to-your-phylogenies.html"><a href="adding-calibration-points-to-your-phylogenies.html#choosing-calibration-points"><i class="fa fa-check"></i><b>20.1</b> Choosing calibration points</a></li>
<li class="chapter" data-level="20.2" data-path="adding-calibration-points-to-your-phylogenies.html"><a href="adding-calibration-points-to-your-phylogenies.html#adding-calibration-points-to-internal-nodes-for-ml-trees"><i class="fa fa-check"></i><b>20.2</b> Adding calibration points to internal nodes for ML trees</a></li>
<li class="chapter" data-level="20.3" data-path="adding-calibration-points-to-your-phylogenies.html"><a href="adding-calibration-points-to-your-phylogenies.html#adding-calibration-points-for-bayesian-trees"><i class="fa fa-check"></i><b>20.3</b> Adding calibration points for Bayesian trees</a></li>
<li class="chapter" data-level="20.4" data-path="adding-calibration-points-to-your-phylogenies.html"><a href="adding-calibration-points-to-your-phylogenies.html#visualizing-the-time-trees-in-figtree"><i class="fa fa-check"></i><b>20.4</b> Visualizing the time trees in FigTree</a></li>
</ul></li>
<li class="part"><span><b>PHYLOGENETIC NETWORKS</b></span></li>
<li class="chapter" data-level="21" data-path="what-is-a-network.html"><a href="what-is-a-network.html"><i class="fa fa-check"></i><b>21</b> What is a network?</a>
<ul>
<li class="chapter" data-level="21.1" data-path="what-is-a-network.html"><a href="what-is-a-network.html#why-do-we-see-competing-phylogenetic-relationships"><i class="fa fa-check"></i><b>21.1</b> Why do we see competing phylogenetic relationships?</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="visualizing-phylogenetic-networks.html"><a href="visualizing-phylogenetic-networks.html"><i class="fa fa-check"></i><b>22</b> Visualizing phylogenetic networks</a>
<ul>
<li class="chapter" data-level="22.1" data-path="visualizing-phylogenetic-networks.html"><a href="visualizing-phylogenetic-networks.html#using-densitree"><i class="fa fa-check"></i><b>22.1</b> Using DensiTree</a></li>
<li class="chapter" data-level="22.2" data-path="visualizing-phylogenetic-networks.html"><a href="visualizing-phylogenetic-networks.html#networks-in-r"><i class="fa fa-check"></i><b>22.2</b> Networks in R</a></li>
</ul></li>
<li class="part"><span><b>ANCESTRAL STATE RECONSTRUCTION</b></span></li>
<li class="chapter" data-level="23" data-path="ancestral-state-reconstruction.html"><a href="ancestral-state-reconstruction.html"><i class="fa fa-check"></i><b>23</b> Ancestral state reconstruction</a>
<ul>
<li class="chapter" data-level="23.1" data-path="ancestral-state-reconstruction.html"><a href="ancestral-state-reconstruction.html#reconstructing-ancestral-sequences"><i class="fa fa-check"></i><b>23.1</b> Reconstructing ancestral sequences</a></li>
<li class="chapter" data-level="23.2" data-path="ancestral-state-reconstruction.html"><a href="ancestral-state-reconstruction.html#reconstructing-ancestral-continuous-phenotypes"><i class="fa fa-check"></i><b>23.2</b> Reconstructing ancestral continuous phenotypes</a></li>
<li class="chapter" data-level="23.3" data-path="ancestral-state-reconstruction.html"><a href="ancestral-state-reconstruction.html#reconstructing-ancestral-discrete-phenotypes"><i class="fa fa-check"></i><b>23.3</b> Reconstructing ancestral discrete phenotypes</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="about-the-authors.html"><a href="about-the-authors.html"><i class="fa fa-check"></i>About the Authors</a></li>
<li class="chapter" data-level="24" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>24</b> References</a></li>
<li class="divider"></li>
<p style="text-align:center;"> <a href="https://github.com/jhudsl/OTTR_Template" target="blank" > This content was published with</a> <a href="https://bookdown.org/" target="blank"> bookdown by: </a> </p>
<p style="text-align:center;"> <a href="http://jhudatascience.org/"> The Johns Hopkins Data Science Lab </a></p>
<a href="http://jhudatascience.org/"><img src="https://jhudatascience.org/images/dasl.png" style=" width: 80%; padding-left: 40px; padding-top: 8px; vertical-align: top "</a>
<p style="text-align:center; font-size: 12px;"> <a href="https://github.com/rstudio4edu/rstudio4edu-book/"> Style adapted from: rstudio4edu-book </a> <a href ="https://creativecommons.org/licenses/by/2.0/"> (CC-BY 2.0) </a></p>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Molecular Phylogenetic Techniques</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<head>
  <meta name="viewport" content="width=device-width,minimum-scale=1.0,maximum-scale=10.0,initial-scale=1.0">
  <!--script src="https://kit.fontawesome.com/6a26f47516.js"></script-->
  <!--<script src="assets/hideOutput.js"></script>-->
  <link href="assets/AnvIL_style/style.css" rel="stylesheet">
</head>
        


<div class="hero-image-container"> 
  <img class= "hero-image" src="assets/AnVIL_style/dasl_thin_main_image.png">
</div>
<div id="downloading-the-sequences-from-genbank" class="section level1" number="9">
<h1><span class="header-section-number">Chapter 9</span> Downloading the sequences from GenBank</h1>
<p>Now that we have identified the sequences for our tree, we need to download those sequences from GenBank into R. One option is to download the sequences directly from GenBank as a fasta file. If you are interested in this option, <a href="http://jonathancrabtree.github.io/Circleator/tutorials/gb_annotation/gb_download.html">here</a> is a good tutorial on how to do it. This will work and the subsequent fasta file can be uploaded into R.</p>
<div id="the-dna.bin-object" class="section level2" number="9.1">
<h2><span class="header-section-number">9.1</span> The DNA.bin object</h2>
<p>However, the library <code>ape</code> has a command that allows us to download sequences from GenBank directly into R and store the sequences as a <code>DNA.bin</code> object. This is a data structure that stores information like DNA sequence, how long each sequence is, information about the species identification of each sequence, and the total base percentages of all the sequences.</p>
<p>The command we’re using is <code>read.GenBank</code>, which takes an argument of the accession number we want to download from GenBank. Since we want to download multiple sequences, We use c(““) to concatenate a string of accession numbers that read.Genbank will interpret.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="downloading-the-sequences-from-genbank.html#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ape) <span class="co">#if you haven&#39;t previously loaded ape</span></span>
<span id="cb23-2"><a href="downloading-the-sequences-from-genbank.html#cb23-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-3"><a href="downloading-the-sequences-from-genbank.html#cb23-3" aria-hidden="true" tabindex="-1"></a><span class="fu">read.GenBank</span>(<span class="fu">c</span>(<span class="st">&quot;JX915632&quot;</span>,<span class="st">&quot;EF105403.1&quot;</span>,<span class="st">&quot;DQ073553.1&quot;</span>,</span>
<span id="cb23-4"><a href="downloading-the-sequences-from-genbank.html#cb23-4" aria-hidden="true" tabindex="-1"></a>      <span class="st">&quot;FJ481575.1&quot;</span>,<span class="st">&quot;EF204545.1&quot;</span>,<span class="st">&quot;AJ314771.1&quot;</span>,<span class="st">&quot;FJ481569.1&quot;</span>,</span>
<span id="cb23-5"><a href="downloading-the-sequences-from-genbank.html#cb23-5" aria-hidden="true" tabindex="-1"></a>      <span class="st">&quot;DQ073533.1&quot;</span>,<span class="st">&quot;AY804128.1&quot;</span>,<span class="st">&quot;AY303125.2&quot;</span>,<span class="st">&quot;KF887414.1&quot;</span>,</span>
<span id="cb23-6"><a href="downloading-the-sequences-from-genbank.html#cb23-6" aria-hidden="true" tabindex="-1"></a>      <span class="st">&quot;D82941.1&quot;</span>,<span class="st">&quot;JX276655.1&quot;</span>))</span></code></pre></div>
<pre><code>## 13 DNA sequences in binary format stored in a list.
## 
## Mean sequence length: 1814.385 
##    Shortest sequence: 1476 
##     Longest sequence: 2490 
## 
## Labels:
## JX915632
## EF105403.1
## DQ073553.1
## FJ481575.1
## EF204545.1
## AJ314771.1
## ...
## 
## Base composition:
##     a     c     g     t 
## 0.310 0.303 0.271 0.115 
## (Total: 23.59 kb)</code></pre>
<p>Now that you have seen what read.Genbank does, we will
save it as an object, and also specify that we want the sequences in ATGC form. When <code>as.character=TRUE</code> is not included (like above), <code>read.GenBank</code> saves all the sequence data in a binary format. Binary is great for computers, but harder for humans to quickly interpret.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="downloading-the-sequences-from-genbank.html#cb25-1" aria-hidden="true" tabindex="-1"></a>grass <span class="ot">&lt;-</span> <span class="fu">read.GenBank</span>(<span class="fu">c</span>(<span class="st">&quot;JX915632&quot;</span>,<span class="st">&quot;EF105403.1&quot;</span>,<span class="st">&quot;DQ073553.1&quot;</span>,</span>
<span id="cb25-2"><a href="downloading-the-sequences-from-genbank.html#cb25-2" aria-hidden="true" tabindex="-1"></a>      <span class="st">&quot;FJ481575.1&quot;</span>,<span class="st">&quot;EF204545.1&quot;</span>,<span class="st">&quot;AJ314771.1&quot;</span>,<span class="st">&quot;FJ481569.1&quot;</span>,</span>
<span id="cb25-3"><a href="downloading-the-sequences-from-genbank.html#cb25-3" aria-hidden="true" tabindex="-1"></a>      <span class="st">&quot;DQ073533.1&quot;</span>,<span class="st">&quot;AY804128.1&quot;</span>,<span class="st">&quot;AY303125.2&quot;</span>,<span class="st">&quot;KF887414.1&quot;</span>,</span>
<span id="cb25-4"><a href="downloading-the-sequences-from-genbank.html#cb25-4" aria-hidden="true" tabindex="-1"></a>      <span class="st">&quot;D82941.1&quot;</span>,<span class="st">&quot;JX276655.1&quot;</span>))</span></code></pre></div>
<div class="notice">
<p>R BASICS</p>
<p>In R, you can do two things with the output of a command. First, you can have the output displayed immediately. This is what you did in the first block of R code above. This can be really helpful if you want to immediately see what your command did, but it’s less helpful if you want to do something with the output. In the first block of code, we managed to download sequences from GenBank and print them to the screen, but we don’t have a way to build trees from the printed screen. (Print in this case refers to the display you see on the R console.)</p>
<p>Second, you can tell R to save the output as an object. This is what we did second block of code with <code>grass &lt;- read.GenBank</code>. The <code>&lt;-</code> operator tells R to redirect the output from <code>read.GenBank</code> to an object (or data structure) named <code>grass</code>. Everything you saw printed on the screen from the first block of code is now saved to <code>grass</code>.</p>
<p>If we ever want to see what objects we have saved in our R session, we can do so by typing the command</p>
<p><code>ls()</code></p>
<p>This tells R to list objects. We can see what each object contains by either typing the name of the object or by using the <code>str</code> (structure) command. The syntax of the structure command is</p>
<p><code>str(object_name)</code></p>
</div>
</div>
<div id="the-fasta-format" class="section level2" number="9.2">
<h2><span class="header-section-number">9.2</span> The fasta format</h2>
<p>While <code>ape</code> and related R packages have no difficulty interpreting a <code>DNA.bin</code> object, other programs need the data in the fasta format. Fasta is a really common format for saving bioinformatic data (probably the most common format used!).</p>
<p>The format itself is quite simple and consists of two lines. The first line is the header line, which starts with &gt;, immediately followed by a unique identifier. The sequence itself is on the second line. The sequence can either the standard <a href="https://www.bioinformatics.org/sms/iupac.html">IUPAC</a> nucleic acid or amino acid codes, with additional characters for gaps (-), uracil (U), or translation stops (*).</p>
<p>The first 60 nucleotides from the <em>Triticum aestivium</em> sequence from above might look like this in fasta format:</p>
<blockquote>
<p>JX915632_Triticum_aestivium
atggctaagcggctggtcctctttgcagcagtagccgtcgccctcgtggctctcaccgcc</p>
</blockquote>
<p>We can convert and save our <code>DNA.bin</code> object in fasta format using a tool from the <code>ape</code> package. The <code>write.dna</code> commands takes three arguments: the first argument tells R the DNA.bin file to use, the second argument says what to name the new file, and the third argument says what format to use for the new file.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="downloading-the-sequences-from-genbank.html#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.dna</span>( grass, <span class="at">file =</span> <span class="st">&#39;grass.fasta&#39;</span>, <span class="at">format =</span> <span class="st">&#39;fasta&#39;</span> )</span></code></pre></div>
<p>If you check your RStudio files (on the lower left side of the screen, you’ll see a tab named Files), you should see a newly-created file called “grass.fasta”. If you open it up, the file will look something like this:</p>
<p><img src="resources/images/04-fasta.png" title="Major point!! example image" alt="Major point!! example image" style="display: block; margin: auto;" /></p>
<p>Going forward, we will use both the <code>DNA.bin</code> object and the fasta file.</p>
<div class="notice">
<p>R BASICS</p>
<p>You might have noticed that we didn’t save the output of <code>write.dna</code> to an object. That’s because any of the <code>write</code> commands are automatically saving the output to a file on your computer (or, on AnVIL, to your persistent disk). The output is saved in what as known as your working directory. You can check what your current working directory is during any R session with the command</p>
<p><code>getwd()</code></p>
</div>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="downloading-the-sequences-from-genbank.html#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.0.2 (2020-06-22)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.3 LTS
## 
## Matrix products: default
## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] ape_5.4-1
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_1.0.8      knitr_1.33      magrittr_2.0.2  hms_0.5.3      
##  [5] lattice_0.20-41 R6_2.4.1        rlang_0.4.10    stringr_1.4.0  
##  [9] tools_4.0.2     parallel_4.0.2  grid_4.0.2      nlme_3.1-149   
## [13] xfun_0.26       jquerylib_0.1.4 htmltools_0.5.0 ellipsis_0.3.1 
## [17] ottrpal_0.1.2   yaml_2.2.1      digest_0.6.25   tibble_3.0.3   
## [21] lifecycle_1.0.0 crayon_1.3.4    bookdown_0.24   readr_1.4.0    
## [25] vctrs_0.3.4     fs_1.5.0        evaluate_0.14   rmarkdown_2.10 
## [29] stringi_1.5.3   compiler_4.0.2  pillar_1.4.6    pkgconfig_2.0.3</code></pre>

</div>
</div>



<hr>
<center> 
  <div class="footer">
      All illustrations <a href="https://creativecommons.org/licenses/by/4.0/">CC-BY. </a>
      <br>
      All other materials <a href= "https://creativecommons.org/licenses/by/4.0/"> CC-BY </a> unless noted otherwise.
  </div>
</center>
            </section>

          </div>
        </div>
      </div>
<a href="finding-sequences-in-genbank.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="why-do-we-align-sequences.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
