<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 16 More Phylogenetics Basics | Molecular Phylogenetic Techniques</title>
  <meta name="description" content="Description about Course/Book." />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 16 More Phylogenetics Basics | Molecular Phylogenetic Techniques" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Description about Course/Book." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 16 More Phylogenetics Basics | Molecular Phylogenetic Techniques" />
  
  <meta name="twitter:description" content="Description about Course/Book." />
  



<meta name="date" content="2025-05-06" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="assets/AnVIL_style/anvil_favicon.ico" type="image/x-icon" />
<link rel="prev" href="building-a-parsimony-tree.html"/>
<link rel="next" href="what-is-maximum-likelihood.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
  
<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-5QVCXY02EY"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-5QVCXY02EY');
</script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #204a87; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #204a87; font-weight: bold; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #ce5c00; font-weight: bold; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="assets/style.css" type="text/css" />
<link rel="stylesheet" href="assets/AnVIL_style/anvil.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<a href="https://anvilproject.org/" target="_blank"><img src="assets/AnVIL_style/logo-anvil.png" style="width: 80%; padding-left: 15px; padding-top: 8px;"</a>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>About this Book</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#skills-level"><i class="fa fa-check"></i>Skills Level</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#anvil-collection"><i class="fa fa-check"></i>AnVIL Collection</a></li>
</ul></li>
<li class="part"><span><b>BOOK OVERVIEW</b></span></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#motivation"><i class="fa fa-check"></i><b>1.1</b> Motivation</a></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#target-audience"><i class="fa fa-check"></i><b>1.2</b> Target Audience</a></li>
<li class="chapter" data-level="1.3" data-path="introduction.html"><a href="introduction.html#curriculum"><i class="fa fa-check"></i><b>1.3</b> Curriculum</a></li>
</ul></li>
<li class="part"><span><b>BEFORE YOU START</b></span></li>
<li class="chapter" data-level="2" data-path="sign-in.html"><a href="sign-in.html"><i class="fa fa-check"></i><b>2</b> Sign in</a></li>
<li class="chapter" data-level="3" data-path="create-a-workspace.html"><a href="create-a-workspace.html"><i class="fa fa-check"></i><b>3</b> Create a workspace</a></li>
<li class="part"><span><b>RUNNING RSTUDIO</b></span></li>
<li class="chapter" data-level="4" data-path="rstudio.html"><a href="rstudio.html"><i class="fa fa-check"></i><b>4</b> Working in RStudio</a>
<ul>
<li class="chapter" data-level="4.1" data-path="rstudio.html"><a href="rstudio.html#launch-rstudio-cloud-environment"><i class="fa fa-check"></i><b>4.1</b> Launch RStudio Cloud Environment</a></li>
<li class="chapter" data-level="4.2" data-path="rstudio.html"><a href="rstudio.html#tour-rstudio"><i class="fa fa-check"></i><b>4.2</b> Tour RStudio</a></li>
<li class="chapter" data-level="4.3" data-path="rstudio.html"><a href="rstudio.html#stopping"><i class="fa fa-check"></i><b>4.3</b> Pause RStudio</a></li>
<li class="chapter" data-level="4.4" data-path="rstudio.html"><a href="rstudio.html#delete-rstudio-cloud-environment"><i class="fa fa-check"></i><b>4.4</b> Delete RStudio Cloud Environment</a></li>
<li class="chapter" data-level="4.5" data-path="rstudio.html"><a href="rstudio.html#video-guide"><i class="fa fa-check"></i><b>4.5</b> Video Guide</a></li>
</ul></li>
<li class="part"><span><b>TREE THINKING</b></span></li>
<li class="chapter" data-level="5" data-path="phylogenetics-basics.html"><a href="phylogenetics-basics.html"><i class="fa fa-check"></i><b>5</b> Phylogenetics basics</a>
<ul>
<li class="chapter" data-level="5.1" data-path="phylogenetics-basics.html"><a href="phylogenetics-basics.html#how-to-read-a-phylogenetic-tree"><i class="fa fa-check"></i><b>5.1</b> How to read a phylogenetic tree</a></li>
<li class="chapter" data-level="5.2" data-path="phylogenetics-basics.html"><a href="phylogenetics-basics.html#outgroups"><i class="fa fa-check"></i><b>5.2</b> Outgroups</a></li>
<li class="chapter" data-level="5.3" data-path="phylogenetics-basics.html"><a href="phylogenetics-basics.html#branch-lengths"><i class="fa fa-check"></i><b>5.3</b> Branch lengths</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="visualizing-trees-in-r.html"><a href="visualizing-trees-in-r.html"><i class="fa fa-check"></i><b>6</b> Visualizing trees in R</a>
<ul>
<li class="chapter" data-level="6.1" data-path="visualizing-trees-in-r.html"><a href="visualizing-trees-in-r.html#creating-a-newick-object"><i class="fa fa-check"></i><b>6.1</b> Creating a Newick object</a></li>
<li class="chapter" data-level="6.2" data-path="visualizing-trees-in-r.html"><a href="visualizing-trees-in-r.html#drawing-trees"><i class="fa fa-check"></i><b>6.2</b> Drawing trees</a></li>
<li class="chapter" data-level="6.3" data-path="visualizing-trees-in-r.html"><a href="visualizing-trees-in-r.html#adding-outgroups"><i class="fa fa-check"></i><b>6.3</b> Adding outgroups</a></li>
<li class="chapter" data-level="6.4" data-path="visualizing-trees-in-r.html"><a href="visualizing-trees-in-r.html#drawing-trees-multiple-ways"><i class="fa fa-check"></i><b>6.4</b> Drawing trees multiple ways</a></li>
<li class="chapter" data-level="6.5" data-path="visualizing-trees-in-r.html"><a href="visualizing-trees-in-r.html#the-phylo-class"><i class="fa fa-check"></i><b>6.5</b> The <code>phylo</code> class</a></li>
</ul></li>
<li class="part"><span><b>GETTING SEQUENCES</b></span></li>
<li class="chapter" data-level="7" data-path="what-sequences-should-i-choose.html"><a href="what-sequences-should-i-choose.html"><i class="fa fa-check"></i><b>7</b> What sequences should I choose?</a></li>
<li class="chapter" data-level="8" data-path="finding-sequences-in-genbank.html"><a href="finding-sequences-in-genbank.html"><i class="fa fa-check"></i><b>8</b> Finding sequences in GenBank</a>
<ul>
<li class="chapter" data-level="8.1" data-path="finding-sequences-in-genbank.html"><a href="finding-sequences-in-genbank.html#identifying-a-query-sequence-on-genbank"><i class="fa fa-check"></i><b>8.1</b> Identifying a query sequence on GenBank</a></li>
<li class="chapter" data-level="8.2" data-path="finding-sequences-in-genbank.html"><a href="finding-sequences-in-genbank.html#blastn"><i class="fa fa-check"></i><b>8.2</b> blastn</a></li>
<li class="chapter" data-level="8.3" data-path="finding-sequences-in-genbank.html"><a href="finding-sequences-in-genbank.html#identifying-an-outgroup"><i class="fa fa-check"></i><b>8.3</b> Identifying an outgroup</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="downloading-the-sequences-from-genbank.html"><a href="downloading-the-sequences-from-genbank.html"><i class="fa fa-check"></i><b>9</b> Downloading the sequences from GenBank</a>
<ul>
<li class="chapter" data-level="9.1" data-path="downloading-the-sequences-from-genbank.html"><a href="downloading-the-sequences-from-genbank.html#the-dna.bin-object"><i class="fa fa-check"></i><b>9.1</b> The DNA.bin object</a></li>
<li class="chapter" data-level="9.2" data-path="downloading-the-sequences-from-genbank.html"><a href="downloading-the-sequences-from-genbank.html#the-fasta-format"><i class="fa fa-check"></i><b>9.2</b> The fasta format</a></li>
</ul></li>
<li class="part"><span><b>ALIGNMENT</b></span></li>
<li class="chapter" data-level="10" data-path="why-do-we-align-sequences.html"><a href="why-do-we-align-sequences.html"><i class="fa fa-check"></i><b>10</b> Why do we align sequences?</a></li>
<li class="chapter" data-level="11" data-path="creating-a-multiple-sequence-alignment.html"><a href="creating-a-multiple-sequence-alignment.html"><i class="fa fa-check"></i><b>11</b> Creating a multiple sequence alignment</a>
<ul>
<li class="chapter" data-level="11.1" data-path="creating-a-multiple-sequence-alignment.html"><a href="creating-a-multiple-sequence-alignment.html#installing-the-decipher-package"><i class="fa fa-check"></i><b>11.1</b> Installing the <code>DECIPHER</code> package</a></li>
<li class="chapter" data-level="11.2" data-path="creating-a-multiple-sequence-alignment.html"><a href="creating-a-multiple-sequence-alignment.html#loading-the-fasta-file"><i class="fa fa-check"></i><b>11.2</b> Loading the fasta file</a></li>
<li class="chapter" data-level="11.3" data-path="creating-a-multiple-sequence-alignment.html"><a href="creating-a-multiple-sequence-alignment.html#creating-an-alignment"><i class="fa fa-check"></i><b>11.3</b> Creating an alignment</a></li>
</ul></li>
<li class="part"><span><b>NEIGHBOR JOINING</b></span></li>
<li class="chapter" data-level="12" data-path="the-neighbor-joining-method-of-tree-estimation.html"><a href="the-neighbor-joining-method-of-tree-estimation.html"><i class="fa fa-check"></i><b>12</b> The neighbor joining method of tree estimation</a>
<ul>
<li class="chapter" data-level="12.1" data-path="the-neighbor-joining-method-of-tree-estimation.html"><a href="the-neighbor-joining-method-of-tree-estimation.html#models-of-molecular-evolution"><i class="fa fa-check"></i><b>12.1</b> Models of molecular evolution</a>
<ul>
<li class="chapter" data-level="12.1.1" data-path="the-neighbor-joining-method-of-tree-estimation.html"><a href="the-neighbor-joining-method-of-tree-estimation.html#jc69"><i class="fa fa-check"></i><b>12.1.1</b> JC69</a></li>
<li class="chapter" data-level="12.1.2" data-path="the-neighbor-joining-method-of-tree-estimation.html"><a href="the-neighbor-joining-method-of-tree-estimation.html#k2p"><i class="fa fa-check"></i><b>12.1.2</b> K2P</a></li>
<li class="chapter" data-level="12.1.3" data-path="the-neighbor-joining-method-of-tree-estimation.html"><a href="the-neighbor-joining-method-of-tree-estimation.html#gtr"><i class="fa fa-check"></i><b>12.1.3</b> GTR</a></li>
<li class="chapter" data-level="12.1.4" data-path="the-neighbor-joining-method-of-tree-estimation.html"><a href="the-neighbor-joining-method-of-tree-estimation.html#other-parameters"><i class="fa fa-check"></i><b>12.1.4</b> Other parameters</a></li>
<li class="chapter" data-level="12.1.5" data-path="the-neighbor-joining-method-of-tree-estimation.html"><a href="the-neighbor-joining-method-of-tree-estimation.html#how-do-i-choose-a-model"><i class="fa fa-check"></i><b>12.1.5</b> How do I choose a model?</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="building-a-neighbor-joining-tree.html"><a href="building-a-neighbor-joining-tree.html"><i class="fa fa-check"></i><b>13</b> Building a neighbor joining tree</a>
<ul>
<li class="chapter" data-level="13.1" data-path="building-a-neighbor-joining-tree.html"><a href="building-a-neighbor-joining-tree.html#the-phydat-object"><i class="fa fa-check"></i><b>13.1</b> The phyDat object</a></li>
<li class="chapter" data-level="13.2" data-path="building-a-neighbor-joining-tree.html"><a href="building-a-neighbor-joining-tree.html#choosing-a-substitution-model"><i class="fa fa-check"></i><b>13.2</b> Choosing a substitution model</a></li>
<li class="chapter" data-level="13.3" data-path="building-a-neighbor-joining-tree.html"><a href="building-a-neighbor-joining-tree.html#building-the-neighbor-joining-tree"><i class="fa fa-check"></i><b>13.3</b> Building the neighbor joining tree</a></li>
<li class="chapter" data-level="13.4" data-path="building-a-neighbor-joining-tree.html"><a href="building-a-neighbor-joining-tree.html#visualizing-the-neighbor-joining-tree"><i class="fa fa-check"></i><b>13.4</b> Visualizing the neighbor joining tree</a></li>
<li class="chapter" data-level="13.5" data-path="building-a-neighbor-joining-tree.html"><a href="building-a-neighbor-joining-tree.html#saving-your-trees"><i class="fa fa-check"></i><b>13.5</b> Saving your trees</a></li>
</ul></li>
<li class="part"><span><b>PARSIMONY</b></span></li>
<li class="chapter" data-level="14" data-path="the-parsimony-method-of-tree-estimation.html"><a href="the-parsimony-method-of-tree-estimation.html"><i class="fa fa-check"></i><b>14</b> The parsimony method of tree estimation</a>
<ul>
<li class="chapter" data-level="14.1" data-path="the-parsimony-method-of-tree-estimation.html"><a href="the-parsimony-method-of-tree-estimation.html#parsimony"><i class="fa fa-check"></i><b>14.1</b> Parsimony</a></li>
<li class="chapter" data-level="14.2" data-path="the-parsimony-method-of-tree-estimation.html"><a href="the-parsimony-method-of-tree-estimation.html#tree-searching-methods"><i class="fa fa-check"></i><b>14.2</b> Tree searching methods</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="the-parsimony-method-of-tree-estimation.html"><a href="the-parsimony-method-of-tree-estimation.html#branch-and-bound"><i class="fa fa-check"></i><b>14.2.1</b> Branch and bound</a></li>
<li class="chapter" data-level="14.2.2" data-path="the-parsimony-method-of-tree-estimation.html"><a href="the-parsimony-method-of-tree-estimation.html#nearest-neighbor-interchange-nni"><i class="fa fa-check"></i><b>14.2.2</b> Nearest Neighbor Interchange (NNI)</a></li>
<li class="chapter" data-level="14.2.3" data-path="the-parsimony-method-of-tree-estimation.html"><a href="the-parsimony-method-of-tree-estimation.html#subtree-pruning-and-regrafting-spr"><i class="fa fa-check"></i><b>14.2.3</b> Subtree Pruning and Regrafting (SPR)</a></li>
<li class="chapter" data-level="14.2.4" data-path="the-parsimony-method-of-tree-estimation.html"><a href="the-parsimony-method-of-tree-estimation.html#star-decomposition"><i class="fa fa-check"></i><b>14.2.4</b> Star decomposition</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="building-a-parsimony-tree.html"><a href="building-a-parsimony-tree.html"><i class="fa fa-check"></i><b>15</b> Building a parsimony tree</a>
<ul>
<li class="chapter" data-level="15.1" data-path="building-a-parsimony-tree.html"><a href="building-a-parsimony-tree.html#editing-the-fasta-file"><i class="fa fa-check"></i><b>15.1</b> Editing the fasta file</a></li>
<li class="chapter" data-level="15.2" data-path="building-a-parsimony-tree.html"><a href="building-a-parsimony-tree.html#inferring-a-parsimony-tree"><i class="fa fa-check"></i><b>15.2</b> Inferring a parsimony tree</a></li>
<li class="chapter" data-level="15.3" data-path="building-a-parsimony-tree.html"><a href="building-a-parsimony-tree.html#parsimony-scores"><i class="fa fa-check"></i><b>15.3</b> Parsimony scores</a></li>
<li class="chapter" data-level="15.4" data-path="building-a-parsimony-tree.html"><a href="building-a-parsimony-tree.html#bootstrapping"><i class="fa fa-check"></i><b>15.4</b> Bootstrapping</a></li>
<li class="chapter" data-level="15.5" data-path="building-a-parsimony-tree.html"><a href="building-a-parsimony-tree.html#saving-files"><i class="fa fa-check"></i><b>15.5</b> Saving files</a></li>
</ul></li>
<li class="part"><span><b>MORE TREE THINKING</b></span></li>
<li class="chapter" data-level="16" data-path="more-phylogenetics-basics.html"><a href="more-phylogenetics-basics.html"><i class="fa fa-check"></i><b>16</b> More Phylogenetics Basics</a>
<ul>
<li class="chapter" data-level="16.1" data-path="more-phylogenetics-basics.html"><a href="more-phylogenetics-basics.html#tree-topology"><i class="fa fa-check"></i><b>16.1</b> Tree topology</a></li>
<li class="chapter" data-level="16.2" data-path="more-phylogenetics-basics.html"><a href="more-phylogenetics-basics.html#outgroups-1"><i class="fa fa-check"></i><b>16.2</b> Outgroups</a></li>
<li class="chapter" data-level="16.3" data-path="more-phylogenetics-basics.html"><a href="more-phylogenetics-basics.html#comparing-trees"><i class="fa fa-check"></i><b>16.3</b> Comparing trees</a></li>
</ul></li>
<li class="part"><span><b>MAXIMUM LIKELIHOOD</b></span></li>
<li class="chapter" data-level="17" data-path="what-is-maximum-likelihood.html"><a href="what-is-maximum-likelihood.html"><i class="fa fa-check"></i><b>17</b> What is Maximum Likelihood?</a></li>
<li class="chapter" data-level="18" data-path="estimating-a-tree-using-maximum-likelihood.html"><a href="estimating-a-tree-using-maximum-likelihood.html"><i class="fa fa-check"></i><b>18</b> Estimating a tree using Maximum Likelihood</a>
<ul>
<li class="chapter" data-level="18.1" data-path="estimating-a-tree-using-maximum-likelihood.html"><a href="estimating-a-tree-using-maximum-likelihood.html#creating-a-starting-tree"><i class="fa fa-check"></i><b>18.1</b> Creating a starting tree</a></li>
<li class="chapter" data-level="18.2" data-path="estimating-a-tree-using-maximum-likelihood.html"><a href="estimating-a-tree-using-maximum-likelihood.html#calculating-likelihood-score"><i class="fa fa-check"></i><b>18.2</b> Calculating likelihood score</a></li>
<li class="chapter" data-level="18.3" data-path="estimating-a-tree-using-maximum-likelihood.html"><a href="estimating-a-tree-using-maximum-likelihood.html#optimizing-the-maximum-likelihood-parameters"><i class="fa fa-check"></i><b>18.3</b> Optimizing the maximum likelihood parameters</a></li>
<li class="chapter" data-level="18.4" data-path="estimating-a-tree-using-maximum-likelihood.html"><a href="estimating-a-tree-using-maximum-likelihood.html#changing-the-model-for-your-ml-search"><i class="fa fa-check"></i><b>18.4</b> Changing the model for your ML search</a></li>
<li class="chapter" data-level="18.5" data-path="estimating-a-tree-using-maximum-likelihood.html"><a href="estimating-a-tree-using-maximum-likelihood.html#bootstrapping-the-ml-tree"><i class="fa fa-check"></i><b>18.5</b> Bootstrapping the ML tree</a></li>
<li class="chapter" data-level="18.6" data-path="estimating-a-tree-using-maximum-likelihood.html"><a href="estimating-a-tree-using-maximum-likelihood.html#saving-your-bootstrap-tree"><i class="fa fa-check"></i><b>18.6</b> Saving your bootstrap tree</a></li>
</ul></li>
<li class="part"><span><b>TIME TREES</b></span></li>
<li class="chapter" data-level="19" data-path="what-is-a-time-tree.html"><a href="what-is-a-time-tree.html"><i class="fa fa-check"></i><b>19</b> What is a time tree?</a>
<ul>
<li class="chapter" data-level="19.1" data-path="what-is-a-time-tree.html"><a href="what-is-a-time-tree.html#molecular-clocks"><i class="fa fa-check"></i><b>19.1</b> Molecular clocks</a></li>
<li class="chapter" data-level="19.2" data-path="what-is-a-time-tree.html"><a href="what-is-a-time-tree.html#calibration-points"><i class="fa fa-check"></i><b>19.2</b> Calibration points</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="adding-calibration-points-to-your-phylogenies.html"><a href="adding-calibration-points-to-your-phylogenies.html"><i class="fa fa-check"></i><b>20</b> Adding calibration points to your phylogenies</a>
<ul>
<li class="chapter" data-level="20.1" data-path="adding-calibration-points-to-your-phylogenies.html"><a href="adding-calibration-points-to-your-phylogenies.html#choosing-calibration-points"><i class="fa fa-check"></i><b>20.1</b> Choosing calibration points</a></li>
<li class="chapter" data-level="20.2" data-path="adding-calibration-points-to-your-phylogenies.html"><a href="adding-calibration-points-to-your-phylogenies.html#adding-calibration-points-to-internal-nodes-for-ml-trees"><i class="fa fa-check"></i><b>20.2</b> Adding calibration points to internal nodes for ML trees</a></li>
<li class="chapter" data-level="20.3" data-path="adding-calibration-points-to-your-phylogenies.html"><a href="adding-calibration-points-to-your-phylogenies.html#adding-calibration-points-for-bayesian-trees"><i class="fa fa-check"></i><b>20.3</b> Adding calibration points for Bayesian trees</a></li>
<li class="chapter" data-level="20.4" data-path="adding-calibration-points-to-your-phylogenies.html"><a href="adding-calibration-points-to-your-phylogenies.html#visualizing-the-time-trees-in-figtree"><i class="fa fa-check"></i><b>20.4</b> Visualizing the time trees in FigTree</a></li>
</ul></li>
<li class="part"><span><b>PHYLOGENETIC NETWORKS</b></span></li>
<li class="chapter" data-level="21" data-path="what-is-a-network.html"><a href="what-is-a-network.html"><i class="fa fa-check"></i><b>21</b> What is a network?</a>
<ul>
<li class="chapter" data-level="21.1" data-path="what-is-a-network.html"><a href="what-is-a-network.html#why-do-we-see-competing-phylogenetic-relationships"><i class="fa fa-check"></i><b>21.1</b> Why do we see competing phylogenetic relationships?</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="visualizing-phylogenetic-networks.html"><a href="visualizing-phylogenetic-networks.html"><i class="fa fa-check"></i><b>22</b> Visualizing phylogenetic networks</a>
<ul>
<li class="chapter" data-level="22.1" data-path="visualizing-phylogenetic-networks.html"><a href="visualizing-phylogenetic-networks.html#using-densitree"><i class="fa fa-check"></i><b>22.1</b> Using DensiTree</a></li>
<li class="chapter" data-level="22.2" data-path="visualizing-phylogenetic-networks.html"><a href="visualizing-phylogenetic-networks.html#networks-in-r"><i class="fa fa-check"></i><b>22.2</b> Networks in R</a></li>
</ul></li>
<li class="part"><span><b>ANCESTRAL STATE RECONSTRUCTION</b></span></li>
<li class="chapter" data-level="23" data-path="ancestral-state-reconstruction.html"><a href="ancestral-state-reconstruction.html"><i class="fa fa-check"></i><b>23</b> Ancestral state reconstruction</a>
<ul>
<li class="chapter" data-level="23.1" data-path="ancestral-state-reconstruction.html"><a href="ancestral-state-reconstruction.html#reconstructing-ancestral-sequences"><i class="fa fa-check"></i><b>23.1</b> Reconstructing ancestral sequences</a></li>
<li class="chapter" data-level="23.2" data-path="ancestral-state-reconstruction.html"><a href="ancestral-state-reconstruction.html#reconstructing-ancestral-continuous-phenotypes"><i class="fa fa-check"></i><b>23.2</b> Reconstructing ancestral continuous phenotypes</a></li>
<li class="chapter" data-level="23.3" data-path="ancestral-state-reconstruction.html"><a href="ancestral-state-reconstruction.html#reconstructing-ancestral-discrete-phenotypes"><i class="fa fa-check"></i><b>23.3</b> Reconstructing ancestral discrete phenotypes</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="about-the-authors.html"><a href="about-the-authors.html"><i class="fa fa-check"></i>About the Authors</a></li>
<li class="chapter" data-level="24" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>24</b> References</a></li>
<li class="divider"></li>
<p style="text-align:center;"> <a href="https://github.com/jhudsl/OTTR_Template" target="blank" > This content was published with</a> <a href="https://bookdown.org/" target="blank"> bookdown by: </a> </p>
<p style="text-align:center;"> <a href="http://jhudatascience.org/"> The Johns Hopkins Data Science Lab </a></p>
<a href="http://jhudatascience.org/"><img src="https://jhudatascience.org/images/dasl.png" style=" width: 80%; padding-left: 40px; padding-top: 8px; vertical-align: top "</a>
<p style="text-align:center; font-size: 12px;"> <a href="https://github.com/rstudio4edu/rstudio4edu-book/"> Style adapted from: rstudio4edu-book </a> <a href ="https://creativecommons.org/licenses/by/2.0/"> (CC-BY 2.0) </a></p>
<p style="padding-left: 40px;"><div class="trapezoid" style = "padding-left: 40px;"><span><a href="https://docs.google.com/forms/d/e/1FAIpQLScrDVb_utm55pmb_SHx-RgELTEbCCWdLea0T3IzS0Oj00GE4w/viewform?usp=pp_url&entry.1565230805=Molecular+Phylogenetic+Techniques"> Click here to provide feedback</a> <img src="assets/itcr_arrow.png" style=" width: 10%" ></span></div></p>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Molecular Phylogenetic Techniques</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<head>
  <meta name="viewport" content="width=device-width,minimum-scale=1.0,maximum-scale=10.0,initial-scale=1.0">
  <!--script src="https://kit.fontawesome.com/6a26f47516.js"></script-->
  <!--<script src="assets/hideOutput.js"></script>-->
</head>
        


<div class="hero-image-container"> 
  <img class= "hero-image" src="assets/AnVIL_style/dasl_thin_main_image.png">
</div>
<div id="more-phylogenetics-basics" class="section level1 hasAnchor" number="16">
<h1><span class="header-section-number">Chapter 16</span> More Phylogenetics Basics<a href="more-phylogenetics-basics.html#more-phylogenetics-basics" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Now that you’ve had a chance to build several phylogenies, let’s spend some time examining what the trees are telling us.</p>
<div id="tree-topology" class="section level2 hasAnchor" number="16.1">
<h2><span class="header-section-number">16.1</span> Tree topology<a href="more-phylogenetics-basics.html#tree-topology" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In an earlier chapter, we talked about nodes and clades. As a reminder, a node is the place where two branches connect. Each node represents a hypothesized most recent common ancestor of the taxa on the tips of the branches.</p>
<p><img src="resources/images/08-phyly_mrca.png" title="Major point!! example image" alt="Major point!! example image" style="display: block; margin: auto;" /></p>
<p>In this tree (the grass neighbor joining tree), the node marked in purple connects all the ingroup samples. (The ingroup for this tree are all grass <em>Glu-1</em> samples.) We can assume the most recent common ancestor (MRCA) of the ingroup would have existed at this node. The ingroup here is a an example of <strong>monophyly</strong>; a monophyletic clade is a clade that contains all the descendants of a particular ancestor.</p>
<p><img src="resources/images/08-phyly_clades.png" title="Major point!! example image" alt="Major point!! example image" style="display: block; margin: auto;" /></p>
<p>There are two other terms that describe clades you might run across. <strong>Paraphyly</strong> describes a group that contains some (but not all) of the descendants of a node; in the figure above, the highlighted clade is paraphyletic with respect to the node marked with a purple dot. <strong>Polyphyly</strong> describes a group that contains both descendants and non-descendants of a node. In the grass example, the paraphyletic group contains both the outgroup taxa and a subset of the ingroup.</p>
</div>
<div id="outgroups-1" class="section level2 hasAnchor" number="16.2">
<h2><span class="header-section-number">16.2</span> Outgroups<a href="more-phylogenetics-basics.html#outgroups-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The choice of the two outgroup taxa (Siberian wild rye D-hordein and barley D-hordein) turned out to be a good decision for the grass tree. First, these two taxa are monophyletic compared to the other samples (that is, they share a recent common ancestor, and the ingroup is monophyletic compared to them). Second, the branch lengths for the two outgroup taxa is similar to the branch lengths of all the ingroup taxa. If the branch lengths for the outgroup taxa are too long, the relationships among all the grass <em>Glu-1</em> samples will be obscured.</p>
<p>However, what if our first outgroup choice hadn’t been quite right? In that case, we could edit the fasta file to remove any samples that we needed.</p>
<p>Let’s pretend the outgroup wasn’t monophyletic and that instead the Siberian wild rye D-hordein sample actually fell within the ingroup. In this case, we could simply remove that sample using the <code>phylotools</code> package.</p>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="more-phylogenetics-basics.html#cb91-1" tabindex="-1"></a><span class="fu">library</span>(phylotools)</span>
<span id="cb91-2"><a href="more-phylogenetics-basics.html#cb91-2" tabindex="-1"></a></span>
<span id="cb91-3"><a href="more-phylogenetics-basics.html#cb91-3" tabindex="-1"></a><span class="fu">rm.sequence.fasta</span>(<span class="at">infile =</span> <span class="st">&quot;grass_aligned-renamed.fasta&quot;</span>, <span class="at">outfile =</span> <span class="st">&quot;sequence.removed.fasta&quot;</span>, <span class="at">to.rm =</span> <span class="st">&quot;Siberian wild rye_D-hordein&quot;</span>)</span></code></pre></div>
<pre><code>## sequence.removed.fasta has been saved to  /__w/AnVIL_Phylogenetic-Techniques/AnVIL_Phylogenetic-Techniques</code></pre>
<p>We can then load the new fasta file and estimate a new neighbor joining tree (or any other type of tree) without the problematic sample.</p>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="more-phylogenetics-basics.html#cb93-1" tabindex="-1"></a><span class="fu">library</span>(phangorn)</span>
<span id="cb93-2"><a href="more-phylogenetics-basics.html#cb93-2" tabindex="-1"></a></span>
<span id="cb93-3"><a href="more-phylogenetics-basics.html#cb93-3" tabindex="-1"></a>grass.new <span class="ot">&lt;-</span> <span class="fu">read.dna</span>(<span class="st">&quot;sequence.removed.fasta&quot;</span>, <span class="at">format=</span><span class="st">&#39;fasta&#39;</span>)</span>
<span id="cb93-4"><a href="more-phylogenetics-basics.html#cb93-4" tabindex="-1"></a></span>
<span id="cb93-5"><a href="more-phylogenetics-basics.html#cb93-5" tabindex="-1"></a>dist.matrix <span class="ot">&lt;-</span> <span class="fu">dist.dna</span>(grass.new, <span class="at">model =</span> <span class="st">&quot;K80&quot;</span>)</span>
<span id="cb93-6"><a href="more-phylogenetics-basics.html#cb93-6" tabindex="-1"></a></span>
<span id="cb93-7"><a href="more-phylogenetics-basics.html#cb93-7" tabindex="-1"></a>tree <span class="ot">&lt;-</span> <span class="fu">NJ</span>(dist.matrix)</span>
<span id="cb93-8"><a href="more-phylogenetics-basics.html#cb93-8" tabindex="-1"></a>tree.root <span class="ot">&lt;-</span> <span class="fu">root</span>(tree, <span class="at">outgroup =</span> <span class="st">&#39;barley_D-hordein&#39;</span>)</span>
<span id="cb93-9"><a href="more-phylogenetics-basics.html#cb93-9" tabindex="-1"></a></span>
<span id="cb93-10"><a href="more-phylogenetics-basics.html#cb93-10" tabindex="-1"></a><span class="fu">plot</span>(tree.root, <span class="at">main =</span> <span class="st">&quot;edited Neighbor Joining&quot;</span>)</span></code></pre></div>
<p><img src="resources/images/08-more-tree-thinking_files/figure-html/unnamed-chunk-3-1.png" width="672" /></p>
<p>You can use the same command to remove multiple outgroup taxa, or to remove an outgroup that is too distant (ie, the branch lengths are too long and including the outgroup is obscuring the relationships among the ingroup samples). If you need to remove all of your outgroup, you can instead try midpoint rooting.</p>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="more-phylogenetics-basics.html#cb94-1" tabindex="-1"></a><span class="fu">rm.sequence.fasta</span>(<span class="at">infile =</span> <span class="st">&quot;grass_aligned-renamed.fasta&quot;</span>, <span class="at">outfile =</span> <span class="st">&quot;no_outgroup.fasta&quot;</span>, <span class="at">to.rm =</span> <span class="fu">c</span>(<span class="st">&quot;barley_D-hordein&quot;</span>, <span class="st">&quot;Siberian wild rye_D-hordein&quot;</span>))</span></code></pre></div>
<pre><code>## no_outgroup.fasta has been saved to  /__w/AnVIL_Phylogenetic-Techniques/AnVIL_Phylogenetic-Techniques</code></pre>
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb96-1"><a href="more-phylogenetics-basics.html#cb96-1" tabindex="-1"></a>grass.no_out<span class="ot">&lt;-</span> <span class="fu">read.dna</span>(<span class="st">&quot;no_outgroup.fasta&quot;</span>, <span class="at">format=</span><span class="st">&#39;fasta&#39;</span>)</span>
<span id="cb96-2"><a href="more-phylogenetics-basics.html#cb96-2" tabindex="-1"></a></span>
<span id="cb96-3"><a href="more-phylogenetics-basics.html#cb96-3" tabindex="-1"></a>dist.matrix <span class="ot">&lt;-</span> <span class="fu">dist.dna</span>(grass.no_out, <span class="at">model =</span> <span class="st">&quot;K80&quot;</span>)</span>
<span id="cb96-4"><a href="more-phylogenetics-basics.html#cb96-4" tabindex="-1"></a></span>
<span id="cb96-5"><a href="more-phylogenetics-basics.html#cb96-5" tabindex="-1"></a>tree.no_out <span class="ot">&lt;-</span> <span class="fu">NJ</span>(dist.matrix)</span>
<span id="cb96-6"><a href="more-phylogenetics-basics.html#cb96-6" tabindex="-1"></a></span>
<span id="cb96-7"><a href="more-phylogenetics-basics.html#cb96-7" tabindex="-1"></a>tree.no_out <span class="ot">&lt;-</span> <span class="fu">midpoint</span>(tree.no_out)</span>
<span id="cb96-8"><a href="more-phylogenetics-basics.html#cb96-8" tabindex="-1"></a></span>
<span id="cb96-9"><a href="more-phylogenetics-basics.html#cb96-9" tabindex="-1"></a><span class="fu">plot</span>(tree.no_out, <span class="at">main =</span> <span class="st">&quot;Neighbor Joining, midpoint rooting&quot;</span>)</span></code></pre></div>
<p><img src="resources/images/08-more-tree-thinking_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
</div>
<div id="comparing-trees" class="section level2 hasAnchor" number="16.3">
<h2><span class="header-section-number">16.3</span> Comparing trees<a href="more-phylogenetics-basics.html#comparing-trees" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>When we are trying to determine if two trees are telling us the same thing about the relationships among our samples (that is, the topologies of the two trees are identical), we might compare the descendants of each node. If the descendants of each node are the same, then we know the topologies are the same (even if the order of the clades are not identical - remember, branches can rotate around nodes). We’ll use the grass phylogenies as an example</p>
<p><img src="resources/images/08-nodes_1.png" title="Major point!! example image" alt="Major point!! example image" style="display: block; margin: auto;" /></p>
<p>First, we can see that node 1 (the node that connects all the ingroup branches) exists in both trees.</p>
<p><img src="resources/images/08-nodes_2.png" title="Major point!! example image" alt="Major point!! example image" style="display: block; margin: auto;" /></p>
<p>The same is true for the node that connects Asiatic grass to the crested wheatgrass/mosquito grass/medusahead rye clade, as well as the nodes resolving the relationships among that that clade.</p>
<p><img src="resources/images/08-nodes_3.png" title="Major point!! example image" alt="Major point!! example image" style="display: block; margin: auto;" /></p>
<p>Moving to the bottom of the tree, we can also locate nodes 5, 6, and 7 in both trees. Each of these nodes connects all the same descendants in each tree.</p>
<p><img src="resources/images/08-nodes_4.png" title="Major point!! example image" alt="Major point!! example image" style="display: block; margin: auto;" /></p>
<p>Finally, we can identify the presence of nodes 8, 9, and 10 in both trees. This is a little trickier to see, because the branches have rotated. But we can find a node that unites wheat, mammoth wild rye, and intermediate wheatgrass (node 8) as well as a node that unites only mammoth wild rye and intermediate wheatgrass (node 9). We also have a separate node that unites wheatgrass and tall wheatgrass (node 10).</p>
<p>Since all of the ingroup nodes are the same, we know the topologies of the neighbor joining and parsimony trees are the same. However, it’s a bit tedious to go through and label each node. Luckily, we can use R to compare topologies more quickly. Open the <code>ape</code> library and load your saved trees into the console.</p>
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="more-phylogenetics-basics.html#cb97-1" tabindex="-1"></a><span class="fu">library</span>(ape)</span>
<span id="cb97-2"><a href="more-phylogenetics-basics.html#cb97-2" tabindex="-1"></a></span>
<span id="cb97-3"><a href="more-phylogenetics-basics.html#cb97-3" tabindex="-1"></a>nj.tree <span class="ot">&lt;-</span> <span class="fu">read.tree</span>(<span class="st">&quot;nj_grass.tre&quot;</span>)</span>
<span id="cb97-4"><a href="more-phylogenetics-basics.html#cb97-4" tabindex="-1"></a>spr.tree <span class="ot">&lt;-</span> <span class="fu">read.tree</span>(<span class="st">&quot;spr_grass.tre&quot;</span>)</span>
<span id="cb97-5"><a href="more-phylogenetics-basics.html#cb97-5" tabindex="-1"></a>nni.tree <span class="ot">&lt;-</span> <span class="fu">read.tree</span>(<span class="st">&quot;nni_grass.tre&quot;</span>)</span></code></pre></div>
<p>The <code>ape</code> package has a very useful <code>all.equal</code> command (you can see more details about it <a href="https://rdrr.io/cran/ape/man/all.equal.phylo.html">here</a>). This command allows us to compare topologies.</p>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="more-phylogenetics-basics.html#cb98-1" tabindex="-1"></a><span class="fu">all.equal</span>(spr.tree, nj.tree, <span class="at">use.edge.length =</span> F)</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<p>The first two arguments are the trees we’d like to compare. In order to compare just the topologies, we also include the argument <code>use.edge.length=F</code>, which tells the <code>all.equal</code> command to ignore branch lengths.</p>
<p>If we want to tell whether trees are completely identical (that is, both the topologies and the branch lengths are the same), we can change the last argument to T. (Alternatively, we could leave the last argument off entirely, as the default setting is for <code>use.edge.length</code> is T.)</p>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="more-phylogenetics-basics.html#cb100-1" tabindex="-1"></a><span class="fu">all.equal</span>(spr.tree, nj.tree, <span class="at">use.edge.length =</span> T)</span></code></pre></div>
<pre><code>## [1] FALSE</code></pre>
<p>Not surprisingly, the branch lengths differ between the neighbor joining and SPR parsimony tree. However, maybe the branch lengths are the same between the two trees we estimated using parsimony.</p>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="more-phylogenetics-basics.html#cb102-1" tabindex="-1"></a><span class="fu">all.equal</span>(spr.tree, nni.tree)</span></code></pre></div>
<pre><code>## [1] FALSE</code></pre>
<p>Well, now we know the two parsimony trees aren’t completely identical. However, what if this is because the topologies aren’t the same?</p>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="more-phylogenetics-basics.html#cb104-1" tabindex="-1"></a><span class="fu">all.equal</span>(spr.tree, nni.tree, <span class="at">use.edge.length =</span> F)</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<p>By running the <code>all.equal</code> command again, we can verify the topologies are the same, so these two trees must differ in just the branch length estimates.</p>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="more-phylogenetics-basics.html#cb106-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.3.2 (2023-10-31)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 22.04.4 LTS
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## time zone: Etc/UTC
## tzcode source: system (glibc)
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] phangorn_2.11.1  phylotools_0.2.2 ape_5.7-1       
## 
## loaded via a namespace (and not attached):
##  [1] sass_0.4.8       utf8_1.2.4       generics_0.1.3   xml2_1.3.6      
##  [5] lattice_0.21-9   stringi_1.8.3    hms_1.1.3        digest_0.6.34   
##  [9] magrittr_2.0.3   grid_4.3.2       evaluate_0.23    timechange_0.3.0
## [13] bookdown_0.41    fastmap_1.1.1    Matrix_1.6-1.1   rprojroot_2.0.4 
## [17] jsonlite_1.8.8   processx_3.8.3   chromote_0.3.1   ps_1.7.6        
## [21] promises_1.2.1   httr_1.4.7       fansi_1.0.6      ottrpal_1.3.0   
## [25] codetools_0.2-19 jquerylib_0.1.4  cli_3.6.2        rlang_1.1.4     
## [29] cachem_1.0.8     yaml_2.3.8       parallel_4.3.2   tools_4.3.2     
## [33] tzdb_0.4.0       dplyr_1.1.4      fastmatch_1.1-4  vctrs_0.6.5     
## [37] R6_2.5.1         lifecycle_1.0.4  lubridate_1.9.3  snakecase_0.11.1
## [41] stringr_1.5.1    janitor_2.2.0    pkgconfig_2.0.3  pillar_1.9.0    
## [45] bslib_0.6.1      later_1.3.2      glue_1.7.0       Rcpp_1.0.12     
## [49] highr_0.11       xfun_0.48        tibble_3.2.1     tidyselect_1.2.0
## [53] knitr_1.48       igraph_2.0.2     nlme_3.1-164     htmltools_0.5.7 
## [57] websocket_1.4.2  rmarkdown_2.25   webshot2_0.1.1   readr_2.1.5     
## [61] compiler_4.3.2   quadprog_1.5-8   askpass_1.2.0    openssl_2.1.1</code></pre>

</div>
</div>



<hr>
<center> 
  <div class="footer">
      All illustrations <a href="https://creativecommons.org/licenses/by/4.0/">CC-BY. </a>
      <br>
      All other materials <a href= "https://creativecommons.org/licenses/by/4.0/"> CC-BY </a> unless noted otherwise.
  </div>
</center>
            </section>

          </div>
        </div>
      </div>
<a href="building-a-parsimony-tree.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="what-is-maximum-likelihood.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
