<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 12 The neighbor joining method of tree estimation | Molecular Phylogenetic Techniques</title>
  <meta name="description" content="Description about Course/Book." />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 12 The neighbor joining method of tree estimation | Molecular Phylogenetic Techniques" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Description about Course/Book." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 12 The neighbor joining method of tree estimation | Molecular Phylogenetic Techniques" />
  
  <meta name="twitter:description" content="Description about Course/Book." />
  



<meta name="date" content="2023-05-24" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="assets/AnVIL_style/anvil_favicon.ico" type="image/x-icon" />
<link rel="prev" href="creating-a-multiple-sequence-alignment.html"/>
<link rel="next" href="building-a-neighbor-joining-tree.html"/>
<script src="libs/header-attrs-2.10/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
  
<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-5QVCXY02EY"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-5QVCXY02EY');
</script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="assets/style.css" type="text/css" />
<link rel="stylesheet" href="assets/AnVIL_style/anvil.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<a href="https://anvilproject.org/" target="_blank"><img src="assets/AnVIL_style/logo-anvil.png" style="width: 80%; padding-left: 15px; padding-top: 8px;"</a>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>About this Book</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#skills-level"><i class="fa fa-check"></i>Skills Level</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#anvil-collection"><i class="fa fa-check"></i>AnVIL Collection</a></li>
</ul></li>
<li class="part"><span><b>BOOK OVERVIEW</b></span></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#motivation"><i class="fa fa-check"></i><b>1.1</b> Motivation</a></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#target-audience"><i class="fa fa-check"></i><b>1.2</b> Target Audience</a></li>
<li class="chapter" data-level="1.3" data-path="introduction.html"><a href="introduction.html#curriculum"><i class="fa fa-check"></i><b>1.3</b> Curriculum</a></li>
</ul></li>
<li class="part"><span><b>BEFORE YOU START</b></span></li>
<li class="chapter" data-level="2" data-path="sign-in.html"><a href="sign-in.html"><i class="fa fa-check"></i><b>2</b> Sign in</a></li>
<li class="chapter" data-level="3" data-path="create-a-workspace.html"><a href="create-a-workspace.html"><i class="fa fa-check"></i><b>3</b> Create a workspace</a></li>
<li class="part"><span><b>RUNNING RSTUDIO</b></span></li>
<li class="chapter" data-level="4" data-path="rstudio.html"><a href="rstudio.html"><i class="fa fa-check"></i><b>4</b> Working in RStudio</a>
<ul>
<li class="chapter" data-level="4.1" data-path="rstudio.html"><a href="rstudio.html#launch-rstudio-cloud-environment"><i class="fa fa-check"></i><b>4.1</b> Launch RStudio Cloud Environment</a></li>
<li class="chapter" data-level="4.2" data-path="rstudio.html"><a href="rstudio.html#tour-rstudio"><i class="fa fa-check"></i><b>4.2</b> Tour RStudio</a></li>
<li class="chapter" data-level="4.3" data-path="rstudio.html"><a href="rstudio.html#stopping"><i class="fa fa-check"></i><b>4.3</b> Pause RStudio</a></li>
<li class="chapter" data-level="4.4" data-path="rstudio.html"><a href="rstudio.html#delete-rstudio-cloud-environment"><i class="fa fa-check"></i><b>4.4</b> Delete RStudio Cloud Environment</a></li>
<li class="chapter" data-level="4.5" data-path="rstudio.html"><a href="rstudio.html#video-guide"><i class="fa fa-check"></i><b>4.5</b> Video Guide</a></li>
</ul></li>
<li class="part"><span><b>TREE THINKING</b></span></li>
<li class="chapter" data-level="5" data-path="phylogenetics-basics.html"><a href="phylogenetics-basics.html"><i class="fa fa-check"></i><b>5</b> Phylogenetics basics</a>
<ul>
<li class="chapter" data-level="5.1" data-path="phylogenetics-basics.html"><a href="phylogenetics-basics.html#how-to-read-a-phylogenetic-tree"><i class="fa fa-check"></i><b>5.1</b> How to read a phylogenetic tree</a></li>
<li class="chapter" data-level="5.2" data-path="phylogenetics-basics.html"><a href="phylogenetics-basics.html#outgroups"><i class="fa fa-check"></i><b>5.2</b> Outgroups</a></li>
<li class="chapter" data-level="5.3" data-path="phylogenetics-basics.html"><a href="phylogenetics-basics.html#branch-lengths"><i class="fa fa-check"></i><b>5.3</b> Branch lengths</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="visualizing-trees-in-r.html"><a href="visualizing-trees-in-r.html"><i class="fa fa-check"></i><b>6</b> Visualizing trees in R</a>
<ul>
<li class="chapter" data-level="6.1" data-path="visualizing-trees-in-r.html"><a href="visualizing-trees-in-r.html#creating-a-newick-object"><i class="fa fa-check"></i><b>6.1</b> Creating a Newick object</a></li>
<li class="chapter" data-level="6.2" data-path="visualizing-trees-in-r.html"><a href="visualizing-trees-in-r.html#drawing-trees"><i class="fa fa-check"></i><b>6.2</b> Drawing trees</a></li>
<li class="chapter" data-level="6.3" data-path="visualizing-trees-in-r.html"><a href="visualizing-trees-in-r.html#adding-outgroups"><i class="fa fa-check"></i><b>6.3</b> Adding outgroups</a></li>
<li class="chapter" data-level="6.4" data-path="visualizing-trees-in-r.html"><a href="visualizing-trees-in-r.html#drawing-trees-multiple-ways"><i class="fa fa-check"></i><b>6.4</b> Drawing trees multiple ways</a></li>
<li class="chapter" data-level="6.5" data-path="visualizing-trees-in-r.html"><a href="visualizing-trees-in-r.html#the-phylo-class"><i class="fa fa-check"></i><b>6.5</b> The <code>phylo</code> class</a></li>
</ul></li>
<li class="part"><span><b>GETTING SEQUENCES</b></span></li>
<li class="chapter" data-level="7" data-path="what-sequences-should-i-choose.html"><a href="what-sequences-should-i-choose.html"><i class="fa fa-check"></i><b>7</b> What sequences should I choose?</a></li>
<li class="chapter" data-level="8" data-path="finding-sequences-in-genbank.html"><a href="finding-sequences-in-genbank.html"><i class="fa fa-check"></i><b>8</b> Finding sequences in GenBank</a>
<ul>
<li class="chapter" data-level="8.1" data-path="finding-sequences-in-genbank.html"><a href="finding-sequences-in-genbank.html#identifying-a-query-sequence-on-genbank"><i class="fa fa-check"></i><b>8.1</b> Identifying a query sequence on GenBank</a></li>
<li class="chapter" data-level="8.2" data-path="finding-sequences-in-genbank.html"><a href="finding-sequences-in-genbank.html#blastn"><i class="fa fa-check"></i><b>8.2</b> blastn</a></li>
<li class="chapter" data-level="8.3" data-path="finding-sequences-in-genbank.html"><a href="finding-sequences-in-genbank.html#identifying-an-outgroup"><i class="fa fa-check"></i><b>8.3</b> Identifying an outgroup</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="downloading-the-sequences-from-genbank.html"><a href="downloading-the-sequences-from-genbank.html"><i class="fa fa-check"></i><b>9</b> Downloading the sequences from GenBank</a>
<ul>
<li class="chapter" data-level="9.1" data-path="downloading-the-sequences-from-genbank.html"><a href="downloading-the-sequences-from-genbank.html#the-dna.bin-object"><i class="fa fa-check"></i><b>9.1</b> The DNA.bin object</a></li>
<li class="chapter" data-level="9.2" data-path="downloading-the-sequences-from-genbank.html"><a href="downloading-the-sequences-from-genbank.html#the-fasta-format"><i class="fa fa-check"></i><b>9.2</b> The fasta format</a></li>
</ul></li>
<li class="part"><span><b>ALIGNMENT</b></span></li>
<li class="chapter" data-level="10" data-path="why-do-we-align-sequences.html"><a href="why-do-we-align-sequences.html"><i class="fa fa-check"></i><b>10</b> Why do we align sequences?</a></li>
<li class="chapter" data-level="11" data-path="creating-a-multiple-sequence-alignment.html"><a href="creating-a-multiple-sequence-alignment.html"><i class="fa fa-check"></i><b>11</b> Creating a multiple sequence alignment</a>
<ul>
<li class="chapter" data-level="11.1" data-path="creating-a-multiple-sequence-alignment.html"><a href="creating-a-multiple-sequence-alignment.html#installing-the-decipher-package"><i class="fa fa-check"></i><b>11.1</b> Installing the <code>DECIPHER</code> package</a></li>
<li class="chapter" data-level="11.2" data-path="creating-a-multiple-sequence-alignment.html"><a href="creating-a-multiple-sequence-alignment.html#loading-the-fasta-file"><i class="fa fa-check"></i><b>11.2</b> Loading the fasta file</a></li>
<li class="chapter" data-level="11.3" data-path="creating-a-multiple-sequence-alignment.html"><a href="creating-a-multiple-sequence-alignment.html#creating-an-alignment"><i class="fa fa-check"></i><b>11.3</b> Creating an alignment</a></li>
</ul></li>
<li class="part"><span><b>NEIGHBOR JOINING</b></span></li>
<li class="chapter" data-level="12" data-path="the-neighbor-joining-method-of-tree-estimation.html"><a href="the-neighbor-joining-method-of-tree-estimation.html"><i class="fa fa-check"></i><b>12</b> The neighbor joining method of tree estimation</a>
<ul>
<li class="chapter" data-level="12.1" data-path="the-neighbor-joining-method-of-tree-estimation.html"><a href="the-neighbor-joining-method-of-tree-estimation.html#models-of-molecular-evolution"><i class="fa fa-check"></i><b>12.1</b> Models of molecular evolution</a>
<ul>
<li class="chapter" data-level="12.1.1" data-path="the-neighbor-joining-method-of-tree-estimation.html"><a href="the-neighbor-joining-method-of-tree-estimation.html#jc69"><i class="fa fa-check"></i><b>12.1.1</b> JC69</a></li>
<li class="chapter" data-level="12.1.2" data-path="the-neighbor-joining-method-of-tree-estimation.html"><a href="the-neighbor-joining-method-of-tree-estimation.html#k2p"><i class="fa fa-check"></i><b>12.1.2</b> K2P</a></li>
<li class="chapter" data-level="12.1.3" data-path="the-neighbor-joining-method-of-tree-estimation.html"><a href="the-neighbor-joining-method-of-tree-estimation.html#gtr"><i class="fa fa-check"></i><b>12.1.3</b> GTR</a></li>
<li class="chapter" data-level="12.1.4" data-path="the-neighbor-joining-method-of-tree-estimation.html"><a href="the-neighbor-joining-method-of-tree-estimation.html#other-parameters"><i class="fa fa-check"></i><b>12.1.4</b> Other parameters</a></li>
<li class="chapter" data-level="12.1.5" data-path="the-neighbor-joining-method-of-tree-estimation.html"><a href="the-neighbor-joining-method-of-tree-estimation.html#how-do-i-choose-a-model"><i class="fa fa-check"></i><b>12.1.5</b> How do I choose a model?</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="building-a-neighbor-joining-tree.html"><a href="building-a-neighbor-joining-tree.html"><i class="fa fa-check"></i><b>13</b> Building a neighbor joining tree</a>
<ul>
<li class="chapter" data-level="13.1" data-path="building-a-neighbor-joining-tree.html"><a href="building-a-neighbor-joining-tree.html#the-phydat-object"><i class="fa fa-check"></i><b>13.1</b> The phyDat object</a></li>
<li class="chapter" data-level="13.2" data-path="building-a-neighbor-joining-tree.html"><a href="building-a-neighbor-joining-tree.html#choosing-a-substitution-model"><i class="fa fa-check"></i><b>13.2</b> Choosing a substitution model</a></li>
<li class="chapter" data-level="13.3" data-path="building-a-neighbor-joining-tree.html"><a href="building-a-neighbor-joining-tree.html#building-the-neighbor-joining-tree"><i class="fa fa-check"></i><b>13.3</b> Building the neighbor joining tree</a></li>
<li class="chapter" data-level="13.4" data-path="building-a-neighbor-joining-tree.html"><a href="building-a-neighbor-joining-tree.html#visualizing-the-neighbor-joining-tree"><i class="fa fa-check"></i><b>13.4</b> Visualizing the neighbor joining tree</a></li>
<li class="chapter" data-level="13.5" data-path="building-a-neighbor-joining-tree.html"><a href="building-a-neighbor-joining-tree.html#saving-your-trees"><i class="fa fa-check"></i><b>13.5</b> Saving your trees</a></li>
</ul></li>
<li class="part"><span><b>PARSIMONY</b></span></li>
<li class="chapter" data-level="14" data-path="the-parsimony-method-of-tree-estimation.html"><a href="the-parsimony-method-of-tree-estimation.html"><i class="fa fa-check"></i><b>14</b> The parsimony method of tree estimation</a>
<ul>
<li class="chapter" data-level="14.1" data-path="the-parsimony-method-of-tree-estimation.html"><a href="the-parsimony-method-of-tree-estimation.html#parsimony"><i class="fa fa-check"></i><b>14.1</b> Parsimony</a></li>
<li class="chapter" data-level="14.2" data-path="the-parsimony-method-of-tree-estimation.html"><a href="the-parsimony-method-of-tree-estimation.html#tree-searching-methods"><i class="fa fa-check"></i><b>14.2</b> Tree searching methods</a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="the-parsimony-method-of-tree-estimation.html"><a href="the-parsimony-method-of-tree-estimation.html#branch-and-bound"><i class="fa fa-check"></i><b>14.2.1</b> Branch and bound</a></li>
<li class="chapter" data-level="14.2.2" data-path="the-parsimony-method-of-tree-estimation.html"><a href="the-parsimony-method-of-tree-estimation.html#nearest-neighbor-interchange-nni"><i class="fa fa-check"></i><b>14.2.2</b> Nearest Neighbor Interchange (NNI)</a></li>
<li class="chapter" data-level="14.2.3" data-path="the-parsimony-method-of-tree-estimation.html"><a href="the-parsimony-method-of-tree-estimation.html#subtree-pruning-and-regrafting-spr"><i class="fa fa-check"></i><b>14.2.3</b> Subtree Pruning and Regrafting (SPR)</a></li>
<li class="chapter" data-level="14.2.4" data-path="the-parsimony-method-of-tree-estimation.html"><a href="the-parsimony-method-of-tree-estimation.html#star-decomposition"><i class="fa fa-check"></i><b>14.2.4</b> Star decomposition</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="building-a-parsimony-tree.html"><a href="building-a-parsimony-tree.html"><i class="fa fa-check"></i><b>15</b> Building a parsimony tree</a>
<ul>
<li class="chapter" data-level="15.1" data-path="building-a-parsimony-tree.html"><a href="building-a-parsimony-tree.html#editing-the-fasta-file"><i class="fa fa-check"></i><b>15.1</b> Editing the fasta file</a></li>
<li class="chapter" data-level="15.2" data-path="building-a-parsimony-tree.html"><a href="building-a-parsimony-tree.html#inferring-a-parsimony-tree"><i class="fa fa-check"></i><b>15.2</b> Inferring a parsimony tree</a></li>
<li class="chapter" data-level="15.3" data-path="building-a-parsimony-tree.html"><a href="building-a-parsimony-tree.html#parsimony-scores"><i class="fa fa-check"></i><b>15.3</b> Parsimony scores</a></li>
<li class="chapter" data-level="15.4" data-path="building-a-parsimony-tree.html"><a href="building-a-parsimony-tree.html#bootstrapping"><i class="fa fa-check"></i><b>15.4</b> Bootstrapping</a></li>
<li class="chapter" data-level="15.5" data-path="building-a-parsimony-tree.html"><a href="building-a-parsimony-tree.html#saving-files"><i class="fa fa-check"></i><b>15.5</b> Saving files</a></li>
</ul></li>
<li class="part"><span><b>MORE TREE THINKING</b></span></li>
<li class="chapter" data-level="16" data-path="more-phylogenetics-basics.html"><a href="more-phylogenetics-basics.html"><i class="fa fa-check"></i><b>16</b> More Phylogenetics Basics</a>
<ul>
<li class="chapter" data-level="16.1" data-path="more-phylogenetics-basics.html"><a href="more-phylogenetics-basics.html#tree-topology"><i class="fa fa-check"></i><b>16.1</b> Tree topology</a></li>
<li class="chapter" data-level="16.2" data-path="more-phylogenetics-basics.html"><a href="more-phylogenetics-basics.html#outgroups-1"><i class="fa fa-check"></i><b>16.2</b> Outgroups</a></li>
<li class="chapter" data-level="16.3" data-path="more-phylogenetics-basics.html"><a href="more-phylogenetics-basics.html#comparing-trees"><i class="fa fa-check"></i><b>16.3</b> Comparing trees</a></li>
</ul></li>
<li class="part"><span><b>MAXIMUM LIKELIHOOD</b></span></li>
<li class="chapter" data-level="17" data-path="what-is-maximum-likelihood.html"><a href="what-is-maximum-likelihood.html"><i class="fa fa-check"></i><b>17</b> What is Maximum Likelihood?</a></li>
<li class="chapter" data-level="18" data-path="estimating-a-tree-using-maximum-likelihood.html"><a href="estimating-a-tree-using-maximum-likelihood.html"><i class="fa fa-check"></i><b>18</b> Estimating a tree using Maximum Likelihood</a>
<ul>
<li class="chapter" data-level="18.1" data-path="estimating-a-tree-using-maximum-likelihood.html"><a href="estimating-a-tree-using-maximum-likelihood.html#creating-a-starting-tree"><i class="fa fa-check"></i><b>18.1</b> Creating a starting tree</a></li>
<li class="chapter" data-level="18.2" data-path="estimating-a-tree-using-maximum-likelihood.html"><a href="estimating-a-tree-using-maximum-likelihood.html#calculating-likelihood-score"><i class="fa fa-check"></i><b>18.2</b> Calculating likelihood score</a></li>
<li class="chapter" data-level="18.3" data-path="estimating-a-tree-using-maximum-likelihood.html"><a href="estimating-a-tree-using-maximum-likelihood.html#optimizing-the-maximum-likelihood-parameters"><i class="fa fa-check"></i><b>18.3</b> Optimizing the maximum likelihood parameters</a></li>
<li class="chapter" data-level="18.4" data-path="estimating-a-tree-using-maximum-likelihood.html"><a href="estimating-a-tree-using-maximum-likelihood.html#changing-the-model-for-your-ml-search"><i class="fa fa-check"></i><b>18.4</b> Changing the model for your ML search</a></li>
<li class="chapter" data-level="18.5" data-path="estimating-a-tree-using-maximum-likelihood.html"><a href="estimating-a-tree-using-maximum-likelihood.html#bootstrapping-the-ml-tree"><i class="fa fa-check"></i><b>18.5</b> Bootstrapping the ML tree</a></li>
<li class="chapter" data-level="18.6" data-path="estimating-a-tree-using-maximum-likelihood.html"><a href="estimating-a-tree-using-maximum-likelihood.html#saving-your-bootstrap-tree"><i class="fa fa-check"></i><b>18.6</b> Saving your bootstrap tree</a></li>
</ul></li>
<li class="part"><span><b>TIME TREES</b></span></li>
<li class="chapter" data-level="19" data-path="what-is-a-time-tree.html"><a href="what-is-a-time-tree.html"><i class="fa fa-check"></i><b>19</b> What is a time tree?</a>
<ul>
<li class="chapter" data-level="19.1" data-path="what-is-a-time-tree.html"><a href="what-is-a-time-tree.html#molecular-clocks"><i class="fa fa-check"></i><b>19.1</b> Molecular clocks</a></li>
<li class="chapter" data-level="19.2" data-path="what-is-a-time-tree.html"><a href="what-is-a-time-tree.html#calibration-points"><i class="fa fa-check"></i><b>19.2</b> Calibration points</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="adding-calibration-points-to-your-phylogenies.html"><a href="adding-calibration-points-to-your-phylogenies.html"><i class="fa fa-check"></i><b>20</b> Adding calibration points to your phylogenies</a>
<ul>
<li class="chapter" data-level="20.1" data-path="adding-calibration-points-to-your-phylogenies.html"><a href="adding-calibration-points-to-your-phylogenies.html#choosing-calibration-points"><i class="fa fa-check"></i><b>20.1</b> Choosing calibration points</a></li>
<li class="chapter" data-level="20.2" data-path="adding-calibration-points-to-your-phylogenies.html"><a href="adding-calibration-points-to-your-phylogenies.html#adding-calibration-points-to-internal-nodes-for-ml-trees"><i class="fa fa-check"></i><b>20.2</b> Adding calibration points to internal nodes for ML trees</a></li>
<li class="chapter" data-level="20.3" data-path="adding-calibration-points-to-your-phylogenies.html"><a href="adding-calibration-points-to-your-phylogenies.html#adding-calibration-points-for-bayesian-trees"><i class="fa fa-check"></i><b>20.3</b> Adding calibration points for Bayesian trees</a></li>
<li class="chapter" data-level="20.4" data-path="adding-calibration-points-to-your-phylogenies.html"><a href="adding-calibration-points-to-your-phylogenies.html#visualizing-the-time-trees-in-figtree"><i class="fa fa-check"></i><b>20.4</b> Visualizing the time trees in FigTree</a></li>
</ul></li>
<li class="part"><span><b>PHYLOGENETIC NETWORKS</b></span></li>
<li class="chapter" data-level="21" data-path="what-is-a-network.html"><a href="what-is-a-network.html"><i class="fa fa-check"></i><b>21</b> What is a network?</a>
<ul>
<li class="chapter" data-level="21.1" data-path="what-is-a-network.html"><a href="what-is-a-network.html#why-do-we-see-competing-phylogenetic-relationships"><i class="fa fa-check"></i><b>21.1</b> Why do we see competing phylogenetic relationships?</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="visualizing-phylogenetic-networks.html"><a href="visualizing-phylogenetic-networks.html"><i class="fa fa-check"></i><b>22</b> Visualizing phylogenetic networks</a>
<ul>
<li class="chapter" data-level="22.1" data-path="visualizing-phylogenetic-networks.html"><a href="visualizing-phylogenetic-networks.html#using-densitree"><i class="fa fa-check"></i><b>22.1</b> Using DensiTree</a></li>
<li class="chapter" data-level="22.2" data-path="visualizing-phylogenetic-networks.html"><a href="visualizing-phylogenetic-networks.html#networks-in-r"><i class="fa fa-check"></i><b>22.2</b> Networks in R</a></li>
</ul></li>
<li class="part"><span><b>ANCESTRAL STATE RECONSTRUCTION</b></span></li>
<li class="chapter" data-level="23" data-path="ancestral-state-reconstruction.html"><a href="ancestral-state-reconstruction.html"><i class="fa fa-check"></i><b>23</b> Ancestral state reconstruction</a>
<ul>
<li class="chapter" data-level="23.1" data-path="ancestral-state-reconstruction.html"><a href="ancestral-state-reconstruction.html#reconstructing-ancestral-sequences"><i class="fa fa-check"></i><b>23.1</b> Reconstructing ancestral sequences</a></li>
<li class="chapter" data-level="23.2" data-path="ancestral-state-reconstruction.html"><a href="ancestral-state-reconstruction.html#reconstructing-ancestral-continuous-phenotypes"><i class="fa fa-check"></i><b>23.2</b> Reconstructing ancestral continuous phenotypes</a></li>
<li class="chapter" data-level="23.3" data-path="ancestral-state-reconstruction.html"><a href="ancestral-state-reconstruction.html#reconstructing-ancestral-discrete-phenotypes"><i class="fa fa-check"></i><b>23.3</b> Reconstructing ancestral discrete phenotypes</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="about-the-authors.html"><a href="about-the-authors.html"><i class="fa fa-check"></i>About the Authors</a></li>
<li class="chapter" data-level="24" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>24</b> References</a></li>
<li class="divider"></li>
<p style="text-align:center;"> <a href="https://github.com/jhudsl/OTTR_Template" target="blank" > This content was published with</a> <a href="https://bookdown.org/" target="blank"> bookdown by: </a> </p>
<p style="text-align:center;"> <a href="http://jhudatascience.org/"> The Johns Hopkins Data Science Lab </a></p>
<a href="http://jhudatascience.org/"><img src="https://jhudatascience.org/images/dasl.png" style=" width: 80%; padding-left: 40px; padding-top: 8px; vertical-align: top "</a>
<p style="text-align:center; font-size: 12px;"> <a href="https://github.com/rstudio4edu/rstudio4edu-book/"> Style adapted from: rstudio4edu-book </a> <a href ="https://creativecommons.org/licenses/by/2.0/"> (CC-BY 2.0) </a></p>
<p style="padding-left: 40px;"><div class="trapezoid" style = "padding-left: 40px;"><span>  <a href="https://forms.gle/AK12iVXTjsB7yCUUA"> Click here to provide feedback</a> <img src="assets/itcr_arrow.png" style=" width: 10%" ></span></div></p>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Molecular Phylogenetic Techniques</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<head>
  <meta name="viewport" content="width=device-width,minimum-scale=1.0,maximum-scale=10.0,initial-scale=1.0">
  <!--script src="https://kit.fontawesome.com/6a26f47516.js"></script-->
  <!--<script src="assets/hideOutput.js"></script>-->
</head>
        


<div class="hero-image-container"> 
  <img class= "hero-image" src="assets/AnVIL_style/dasl_thin_main_image.png">
</div>
<div id="the-neighbor-joining-method-of-tree-estimation" class="section level1" number="12">
<h1><span class="header-section-number">Chapter 12</span> The neighbor joining method of tree estimation</h1>
<p>There are two basic types of tree estimation methods: distance methods (which includes neighbor joining, discussed here) and tree-searching methods (which we will cover in later sections). The first phylogenetic trees were built using distance methods. Strictly speaking, these methods can be considered phenetic methods, not phylogenetic methods, since these methods group taxa based on similarity instead of attempting to find the most likely tree. For all practical purposes, though, we can include neighbor joining in under the umbrella of phylogenetics.</p>
<p>In distance-based methods like neighbor joining, the genetic data is converted into a distance matrix, which is used to group taxa based on the genetic distance between them. Neighbor joining specifically works via star decomposition. Let’s take a look at an example (adapted from (Fred Opperdoes)[<a href="http://www.deduveinstitute.be/~opperd/private/neighbor.html" class="uri">http://www.deduveinstitute.be/~opperd/private/neighbor.html</a>]).</p>
<p>We have 5 taxa (A,B,C,D, and E) for which we would like to infer a phylogeny. We’ve calculated the following distances among all the taxa:</p>
<p><img src="resources/images/06-distance_1.png" title="Major point!! example image" alt="Major point!! example image" style="display: block; margin: auto;" /></p>
<p>The shortest distance between any taxa is 13, which means that A and B are most closely related to each other and will form our first group. We begin with all the taxa forming a star shape, and then we add a branch that connects A and B.</p>
<p><img src="resources/images/06-nj_1.png" title="Major point!! example image" alt="Major point!! example image" style="display: block; margin: auto;" /></p>
<p>We keep grouping taxa until the entire star has been resolved into bifurcating branches (nodes that connect two, and only two, branches). For our example tree, we eventually end up with this tree:</p>
<p><img src="resources/images/06-nj_2.png" title="Major point!! example image" alt="Major point!! example image" style="display: block; margin: auto;" /></p>
<p>The neighbor joining method is very fast, requires very few computational resources, and is statistically consistent. It will also always result in only a single tree topology. Unfortunately, many people will become overly confident that the single neighbor joining tree is the true tree, which is not necessarily true. Nowadays, most researchers might start their analysis with a neighbor joining tree, but they generally go on to infer trees using one of the other methods.</p>
<div class="dictionary">
<p>There is another type of distance method you might hear about called UPGMA (unweighted pair group method with arithmetic mean). This is a hierarchical clustering method. However, the tree built using UPGMA depends greatly on the order in which the taxa are added, unlike the neighbor joining method (which will result in the same tree no matter what order the samples are added). As a result, the UPGMA method is almost never used in phylogenetics today.</p>
</div>
<div id="models-of-molecular-evolution" class="section level2" number="12.1">
<h2><span class="header-section-number">12.1</span> Models of molecular evolution</h2>
<p>In the example above, we kind of glossed over the process of converting DNA sequence into genetic distances. In its simplest form, every mutation is given a value, and the total value of all mutations is then used to calculate a genetic distance.</p>
<p>But wait, you might be thinking. How do researchers decide what value to assign each mutation? Are all mutations given the same value, or do some changes “cost” more than others? Aren’t some mutations considered more unlikely than others? These are all excellent questions that should be considered whenever dealing with phylogenetic analysis. Luckily, we don’t have to answer all these questions ourselves. Instead, we can look at published <em>models of molecular evolution</em> (also called substitution models) and decide which model fits our data the best.</p>
<div id="jc69" class="section level3" number="12.1.1">
<h3><span class="header-section-number">12.1.1</span> JC69</h3>
<p>A model of molecular evolution is a set of rules that determines how much each mutation costs relative to all other mutations. One of the earliest published models is the JC69 model (sometimes written as JC), proposed by Jukes and Cantor in 1969. The JC69 model assumes equal base frequencies (that is, each nucleotide makes up 25% of the bases in the DNA sequence), as well as equal mutation rates. A mutation of an A to a T is the same as the mutation as a T to an A or an A to a C. As a result, the JC69 model has only 1 parameter - the mutation rate.</p>
</div>
<div id="k2p" class="section level3" number="12.1.2">
<h3><span class="header-section-number">12.1.2</span> K2P</h3>
<p>Many other models of molecular evolution have been published since JC69. One of the more popular ones is the Kimura two-parameter model (also called K80, or K2P). This model still assumes equal base frequencies, but allows for a distinct mutation rate for transitions (mutations between purines or pyrimidines) and a different mutation rate for tranversions (mutations from purine to pyrimidine, or vice versa). As you might have guessed from the name, this particular model has two parameters.</p>
</div>
<div id="gtr" class="section level3" number="12.1.3">
<h3><span class="header-section-number">12.1.3</span> GTR</h3>
<p>One of the most complex models you might come across is the general time reversible model, or GTR. This model is extremely flexible but also parameter-heavy. There are no assumptions about base frequencies, and each possible mutation is given its own mutation rate. (A quick note: the mutation rate from A to G is the same as the mutation rate from G to A in this model.) As a result, this model has 9 parameters: 6 mutation rates + 3 base frequencies. (Because the sum of the 4 base frequencies must equal 1, we only have to describe 3 of them in order to calculate all 4. That’s why we only need 3 parameters to model the base frequencies for GTR.)</p>
</div>
<div id="other-parameters" class="section level3" number="12.1.4">
<h3><span class="header-section-number">12.1.4</span> Other parameters</h3>
<p>Sometimes you will run across a model that has “+ I” added to it (like GTR + I). The “I” refers to a invariant sites parameter. In these models, each base has a certain probability of being invariate, or unlikely to mutate.</p>
<p>Another option you might see is using a gamma distribution to estimate how many times any given base in a sequence might mutate. We often think of mutation as a one-and-done process, where a nucleotide might mutate only once. This isn’t a bad assumption; in fact, given that mutation is so rare, it is extremely unlikely for a nucleotide to mutate multiple times, though it does happen (particularly as divergence times between taxa increase). A solution to this dilemma is to include a parameter that describes how many times a particular nucleotide has undergone mutation. A gamma distribution is used because the highest probability is found in the smallest values of x, with a low probability of larger values of x (which would be multiple mutation hits at the same nucleotide). A model using a gamma distribution with have a “+ G”.</p>
<p>Finally, some models with have a “+ SS” added to it. The SS indicates a site-specific variation parameter. These models will estimate different mutation rates for each codon position. Site-specific models are much less common than models that include invariate sites parameters or a $ parameter.</p>
</div>
<div id="how-do-i-choose-a-model" class="section level3" number="12.1.5">
<h3><span class="header-section-number">12.1.5</span> How do I choose a model?</h3>
<p>There are enough substitution models out there that it can be a bit overwhelming to pick one for your phylogenetic analyses. (The models discussed above are in no way an exhaustive list.) Many researchers will have a favorite model they use, while others will rely model test programs to tell them which model of molecular evolution best fits their data. Although GTR + I + $ might be the most biologically realistic model, it is also parameter-heavy and may not be the best choice when dealing with a sequence alignment that includes only a small number of phylogenetically-informative sites.</p>
<pre><code>## Determining distance matrix based on shared 10-mers:
## ================================================================================
## 
## Time difference of 0.01 secs
## 
## Clustering into groups by similarity:
## ================================================================================
## 
## Time difference of 0.01 secs
## 
## Aligning Sequences:
## ================================================================================
## 
## Time difference of 0.43 secs
## 
## Iteration 1 of 2:
## 
## Determining distance matrix based on alignment:
## ================================================================================
## 
## Time difference of 0 secs
## 
## Reclustering into groups by similarity:
## ================================================================================
## 
## Time difference of 0 secs
## 
## Realigning Sequences:
## ================================================================================
## 
## Time difference of 0.63 secs
## 
## Iteration 2 of 2:
## 
## Determining distance matrix based on alignment:
## ================================================================================
## 
## Time difference of 0 secs
## 
## Reclustering into groups by similarity:
## ================================================================================
## 
## Time difference of 0 secs
## 
## Realigning Sequences:
## ================================================================================
## 
## Time difference of 0.95 secs
## 
## Refining the alignment:
## ================================================================================
## 
## Time difference of 0.72 secs</code></pre>
</div>
</div>
</div>
<hr>
<center> 
  <div class="footer">
      All illustrations <a href="https://creativecommons.org/licenses/by/4.0/">CC-BY. </a>
      <br>
      All other materials <a href= "https://creativecommons.org/licenses/by/4.0/"> CC-BY </a> unless noted otherwise.
  </div>
</center>
            </section>

          </div>
        </div>
      </div>
<a href="creating-a-multiple-sequence-alignment.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="building-a-neighbor-joining-tree.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
